<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
        "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
  <meta content="text/html; charset=ISO-8859-1" http-equiv="content-type">
  <title>primer3 release 2.0.0 - manual</title>
  <style type="text/css">
  body {
  background-color:white;
  color:black;
  font-size:100.01%;
  margin:0;
  min-width:41em;
  padding:5px;
  text-align:left;
  }
  td {
  font-size:0.78em;
  }
  h2 {
  padding-top:1.2em;
  }
  h3 {
  padding-top:1.2em;
  }
  div#page {
  background:#FFFFFF;
  margin:25px;
  padding:0;
  text-align:left;
  width:850px;
  }
  </style>
</head>
<body>
<div id="page">


<!-- This file was created by the documentation scripts -->
<!-- Do not edit this file -->
<!-- Edit the documentation XMLs instead! -->


<h2>Primer3web - Help</h2><h2>Index of contents</h2><p>
<a href="#copyrightLicense">1. COPYRIGHT AND LICENSE</a><br>
<a href="#introduction">2. INTRODUCTION</a><br>
<a href="#citationRequest">3. CITING PRIMER3</a><br>
<a href="#pickAdvice">4. ADVICE FOR PICKING PRIMERS</a><br>
<a href="#cautions">5. CAUTIONS</a><br>
<a href="#findNoPrimers">6. WHAT TO DO IF PRIMER3 CANNOT FIND ANY PRIMERS?</a><br>
<a href="#contact">7. REPORTING BUGS AND PROBLEMS AND SUGGESTING ENHANCEMENTS</a><br>
<a href="#sequenceTags">8. "SEQUENCE" INPUT TAGS</a><br>
<a href="#globalTags">9. "GLOBAL" INPUT TAGS</a><br>
<a href="#outputHelp">10. OUTPUT HELP</a><br>
<a href="#providedMisprimingLibs">11. POVIDED MISPRIMING LIBRARIES</a><br>
<a href="#calculatePenalties">12. HOW PRIMER3 CALCULATES THE PENALTY VALUE</a><br>
<a href="#webInterface">13. THE PRIMER3 WWW INTERFACE</a><br>
<a href="#acknowledgments">14. ACKNOWLEDGMENTS</a><br>
</p>
<h2><a name="copyrightLicense">1. COPYRIGHT AND LICENSE</a></h2>

<pre>Copyright (c) 1996,1997,1998,1999,2000,2001,2004,2006,2007,2008
Whitehead Institute for Biomedical Research, Steve Rozen
(http://purl.com/STEVEROZEN/), Andreas Untergasser and Helen Skaletsky All rights
reserved.

    This file is part of the primer3 suite and libraries.

    The primer3 suite and libraries are free software;
    you can redistribute them and/or modify them under the terms
    of the GNU General Public License as published by the Free
    Software Foundation; either version 2 of the License, or (at
    your option) any later version.

    This software is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this software (file gpl-2.0.txt in the source
    distribution); if not, write to the Free Software
    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
</pre>

<h2><a name="introduction">2. INTRODUCTION</a></h2>

<p>Primer3 picks primers for PCR reactions, considering as criteria:
<br>
o oligonucleotide melting temperature, size, GC content,
  and primer-dimer possibilities,<br>
<br>
o PCR product size,<br>
<br>
o positional constraints within the source (template) sequence, and<br>
<br>
o possibilities for ectopic priming (amplifying the wrong sequence)<br>
<br>
o many other constraints.<br>
<br>
All of these criteria are user-specifiable as constraints, and
some are specifiable as terms in an objective function that
characterizes an optimal primer pair.<br>
<br>
Whitehead Institute for Biomedical Research provides a basic
web-based front end to primer3 named primer3web at
<a href="http://fokker.wi.mit.edu/cgi-bin/primer3/primer3_www.cgi">http://fokker.wi.mit.edu/cgi-bin/primer3/primer3_www.cgi</a><br>
<br>
Wageningen University Laboratory of Bioinformatics provides a enhanced 
web-based front end to primer3 named primer3plus at
<a href="http://www.bioinformatics.nl/cgi-bin/primer3plus/primer3plus.cgi">http://www.bioinformatics.nl/cgi-bin/primer3plus/primer3plus.cgi</a></p>

<h2><a name="citationRequest">3. CITING PRIMER3</a></h2>

<p>We request but do not require that use of this software be cited in
publications as<br>
<br>
Steve Rozen and Helen J. Skaletsky (2000) Primer3 on the WWW for
general users and for biologist programmers.
In: Krawetz S, Misener S (eds) Bioinformatics Methods and
Protocols: Methods in Molecular Biology.  Humana Press, Totowa,
NJ, pp 365-386<br>
<br>
The paper is available at
<a href="http://jura.wi.mit.edu/rozen/papers/rozen-and-skaletsky-2000-primer3.pdf">http://jura.wi.mit.edu/rozen/papers/rozen-and-skaletsky-2000-primer3.pdf</a><br>
<br>	
Source code available at <a href="http://sourceforge.net/projects/primer3/">http://sourceforge.net/projects/primer3/</a>.</p>

<h2><a name="pickAdvice">4. ADVICE FOR PICKING PRIMERS</a></h2>

<p>We suggest consulting: Wojciech Rychlik (1993) "Selection of
Primers for Polymerase Chain Reaction" in BA White, Ed., "Methods
in Molecular Biology, Vol. 15: PCR Protocols: Current Methods and
Applications", pp 31-40, Humana Press, Totowa NJ.</p>

<h2><a name="cautions">5. CAUTIONS</a></h2>

<p>Some of the most important issues in primer picking can be
addressed only before using primer3. These are sequence quality 
(including making sure the sequence is not vector and not chimeric) 
and avoiding repetitive elements.<br>
<br>
Techniques for avoiding problems include a thorough understanding
of possible vector contaminants and cloning artifacts coupled
with database searches using blast, Fasta, or other similarity
searching program to screen for vector contaminants and possible
repeats.  Repbase (J. Jurka, A.F.A. Smit, C. Pethiyagoda, and
others, 1995-1996, <a href="ftp://ncbi.nlm.nih.gov/repository/repbase">ftp://ncbi.nlm.nih.gov/repository/repbase</a>)
is an excellent source of repeat sequences and pointers to the
literature.  (The Repbase files need to be converted to Fasta format
before they can be used by primer3.) Primer3 now allows you to screen
candidate oligos against a Mispriming Library (or a Mishyb Library in
the case of internal oligos).<br>
<br>
Sequence quality can be controlled by manual trace viewing and
quality clipping or automatic quality clipping programs.  Low-
quality bases should be changed to N's or can be made part of
Excluded Regions. The beginning of a sequencing read is often
problematic because of primer peaks, and the end of the read
often contains many low-quality or even meaningless called bases.
Therefore, when picking primers from single-pass sequence it is
often best to use the <a href="#SEQUENCE_INCLUDED_REGION">SEQUENCE_INCLUDED_REGION</a> parameter to 
ensure that primer3 chooses primers in the high quality region 
of the read.<br>
<br>
In addition, primer3 takes as input a <a href="#SEQUENCE_QUALITY">SEQUENCE_QUALITY</a> list for
use with those base calling programs (e.g. Phred)
that output this information.</p>

<h2><a name="findNoPrimers">6. WHAT TO DO IF PRIMER3 CANNOT FIND ANY PRIMERS?</a></h2>

<p>Try relaxing various parameters, including the
self-complementarity parameters and max and min oligo melting
temperatures.  For example, for very A-T-rich regions you might
have to increase maximum primer size or decrease minimum melting
temperature.  It is usually unwise to reduce the minimum primer
size if your template is complex (e.g. a mammalian genome), since
small primers are more likely to be non-specific.  Make sure that
there are adequate stretches of non-Ns in the regions in which
you wish to pick primers.  If necessary you can also allow an N
in your primer and use an oligo mixture containing all four bases
at that position.<br>
<br>
Try setting the <a href="#PRIMER_EXPLAIN_FLAG">PRIMER_EXPLAIN_FLAG</a> input tag.</p>

<h2><a name="contact">7. REPORTING BUGS AND PROBLEMS AND SUGGESTING ENHANCEMENTS</a></h2>

<p>For reporting errors
or requesting enhancements, please see 
http://primer3.sourceforge.net/contact.php</p>

<h2><a name="sequenceTags">8. "SEQUENCE" INPUT TAGS</a></h2>

<p>"Sequence" input tags start with SEQUENCE_... and describe a 
particular input sequence to primer3. They are reset after every 
Boulder record. Errors in "Sequence" input tags invalidate the 
current record, but primer3 will continue to process additional 
records.</p>
<table style="text-align: left; width: 800px;" border="1">
	  <tr>
	    <td><a href="#SEQUENCE_EXCLUDED_REGION">SEQUENCE_EXCLUDED_REGION</a></td>
	    <td><a href="#SEQUENCE_INCLUDED_REGION">SEQUENCE_INCLUDED_REGION</a></td>
	    <td><a href="#SEQUENCE_QUALITY">SEQUENCE_QUALITY</a></td>
	  </tr>
	  <tr>
	    <td><a href="#SEQUENCE_FORCE_LEFT_END">SEQUENCE_FORCE_LEFT_END</a></td>
	    <td><a href="#SEQUENCE_INTERNAL_EXCLUDED_REGION">SEQUENCE_INTERNAL_EXCLUDED_REGION</a></td>
	    <td><a href="#SEQUENCE_START_CODON_POSITION">SEQUENCE_START_CODON_POSITION</a></td>
	  </tr>
	  <tr>
	    <td><a href="#SEQUENCE_FORCE_LEFT_START">SEQUENCE_FORCE_LEFT_START</a></td>
	    <td><a href="#SEQUENCE_INTERNAL_OLIGO">SEQUENCE_INTERNAL_OLIGO</a></td>
	    <td><a href="#SEQUENCE_TARGET">SEQUENCE_TARGET</a></td>
	  </tr>
	  <tr>
	    <td><a href="#SEQUENCE_FORCE_RIGHT_END">SEQUENCE_FORCE_RIGHT_END</a></td>
	    <td><a href="#SEQUENCE_PRIMER">SEQUENCE_PRIMER</a></td>
	    <td><a href="#SEQUENCE_TEMPLATE">SEQUENCE_TEMPLATE</a></td>
	  </tr>
	  <tr>
	    <td><a href="#SEQUENCE_FORCE_RIGHT_START">SEQUENCE_FORCE_RIGHT_START</a></td>
	    <td><a href="#SEQUENCE_PRIMER_OVERLAP_POS">SEQUENCE_PRIMER_OVERLAP_POS</a></td>
	    <td>&nbsp;</td>
	  </tr>
	  <tr>
	    <td><a href="#SEQUENCE_ID">SEQUENCE_ID</a></td>
	    <td><a href="#SEQUENCE_PRIMER_REVCOMP">SEQUENCE_PRIMER_REVCOMP</a></td>
	    <td>&nbsp;</td>
	  </tr>
</table>
<h3><a name="SEQUENCE_ID">SEQUENCE_ID (string; default empty)</a></h3>

<p>Short description of the sequence. It is used as an identifier 
that is reproduced in the output to enable users to identify the 
source of the chosen primers.<br>
<br>
This tag must be present if <a href="#P3_FILE_FLAG">P3_FILE_FLAG</a> is non-zero.</p>


<h3><a name="SEQUENCE_TEMPLATE">SEQUENCE_TEMPLATE (nucleotide sequence; default empty)</a></h3>

<p>The sequence from which to choose primers.  The sequence
must be presented 5' -> 3' (i.e, in the normal way).
In general, the bases may be upper or lower case, but 
lower case letters are treated 
specially if <a href="#PRIMER_LOWERCASE_MASKING">PRIMER_LOWERCASE_MASKING</a> is set.
The entire sequence MUST be all on a single line. 
(In other words, the sequence cannot span several lines.)
</p>


<h3><a name="SEQUENCE_INCLUDED_REGION">SEQUENCE_INCLUDED_REGION (interval list; default empty)</a></h3>

<p>A sub-region of the given sequence in which to pick primers.  For
example, often the first dozen or so bases of a sequence are
vector, and should be excluded from consideration. The value for
this parameter has the form</p>
<pre>
<tt><i>&lt;start&gt;</i></tt>,<tt><i>&lt;length&gt;</i></tt>
</pre>
<p>where <tt><i>&lt;start&gt;</i></tt> is the index of the first base to consider,
and <tt><i>&lt;length&gt;</i></tt> is the number of subsequent bases in the
primer-picking region.</p>


<h3><a name="SEQUENCE_TARGET">SEQUENCE_TARGET (interval list; default empty)</a></h3>

<p>If one or more Targets is specified then a legal primer pair must
flank at least one of them.  A Target might be a simple sequence
repeat site (for example a CA repeat) or a single-base-pair
polymorphism.  The value should be a space-separated list of</p>
<pre>
<tt><i>&lt;start&gt;</i></tt>,<tt><i>&lt;length&gt;</i></tt>
</pre>
<p>pairs where <tt><i>&lt;start&gt;</i></tt> is the index of the first base of a
Target, and <tt><i>&lt;length&gt;</i></tt> is its length.</p>


<h3><a name="SEQUENCE_EXCLUDED_REGION">SEQUENCE_EXCLUDED_REGION (interval list; default empty)</a></h3>

<p>Primers and oligos may not overlap any region specified in this tag.
The associated value must be a space-separated list of</p>
<pre>
<tt><i>&lt;start&gt;</i></tt>,<tt><i>&lt;length&gt;</i></tt>
</pre>
<p>pairs where <tt><i>&lt;start&gt;</i></tt> is the index of the first base of
the excluded region, and <tt><i>&lt;length&gt;</i></tt> is its length.  This tag is
useful for tasks such as excluding regions of low sequence
quality or for excluding regions containing repetitive elements
such as ALUs or LINEs.</p>


<h3><a name="SEQUENCE_PRIMER_OVERLAP_POS">SEQUENCE_PRIMER_OVERLAP_POS (quality list; default empty)</a></h3>

<p>A list of space separated integers. If this list is not empty primer3
tries to pick primer pairs were the forward or the reverse primer 
overlaps one of these positions for at least 
<a href="#PRIMER_POS_OVERLAP_TO_END_DIST">PRIMER_POS_OVERLAP_TO_END_DIST</a> nucleotides.</p>


<h3><a name="SEQUENCE_INTERNAL_EXCLUDED_REGION">SEQUENCE_INTERNAL_EXCLUDED_REGION (interval list; default empty)</a></h3>

<p>Middle oligos may not overlap any region specified by this tag.
The associated value must be a space-separated list of</p>
<pre>
<tt><i>&lt;start&gt;</i></tt>,<tt><i>&lt;length&gt;</i></tt>
</pre>
<p>pairs, where &lt;start&gt; is the index of the first base of
an excluded region, and &lt;length&gt; is its length.  Often one would
make Target regions excluded regions for internal oligos.</p>


<h3><a name="SEQUENCE_PRIMER">SEQUENCE_PRIMER (nucleotide sequence; default empty)</a></h3>

<p>The sequence of a left primer to check and around which to design
right primers and optional internal oligos.  Must be a substring
of <a href="#SEQUENCE_TEMPLATE">SEQUENCE_TEMPLATE</a>.</p>


<h3><a name="SEQUENCE_INTERNAL_OLIGO">SEQUENCE_INTERNAL_OLIGO (nucleotide sequence; default empty)</a></h3>

<p>The sequence of an internal oligo to check and around which to
design left and right primers.  Must be a substring of 
<a href="#SEQUENCE_TEMPLATE">SEQUENCE_TEMPLATE</a>.</p>


<h3><a name="SEQUENCE_PRIMER_REVCOMP">SEQUENCE_PRIMER_REVCOMP (nucleotide sequence; default empty)</a></h3>

<p>The sequence of a right primer to check and around which to
design left primers and optional internal oligos.  Must be a
substring of the reverse strand of <a href="#SEQUENCE_TEMPLATE">SEQUENCE_TEMPLATE</a>.</p>


<h3><a name="SEQUENCE_START_CODON_POSITION">SEQUENCE_START_CODON_POSITION (int; default -1000000)</a></h3>

<p>This parameter should be considered EXPERIMENTAL at this point.
Please check the output carefully; some erroneous inputs might
cause an error in primer3.<br>
<br>
Index of the first base of a start codon.  This parameter allows
primer3 to select primer pairs to create in-frame amplicons
e.g. to create a template for a fusion protein.  Primer3 will
attempt to select an in-frame left primer, ideally starting at or
to the left of the start codon, or to the right if necessary.
Negative values of this parameter are legal if the actual start
codon is to the left of available sequence. If this parameter is
non-negative primer3 signals an error if the codon at the
position specified by this parameter is not an ATG.  A value less
than or equal to -10^6 indicates that primer3 should ignore this
parameter.<br>
<br>
Primer3 selects the position of the right primer by scanning
right from the left primer for a stop codon.  Ideally the right
primer will end at or after the stop codon.</p>


<h3><a name="SEQUENCE_QUALITY">SEQUENCE_QUALITY (quality list; default empty)</a></h3>

<p>A list of space separated integers. There must be exactly
one integer for each base in <a href="#SEQUENCE_TEMPLATE">SEQUENCE_TEMPLATE</a> if this argument is
non-empty.  For example, for the sequence ANNTTCA...
<a href="#SEQUENCE_QUALITY">SEQUENCE_QUALITY</a> might be 45 10 0 50 30 34 50 67 ....
High numbers indicate high confidence in the base called at
that position and low numbers indicate low confidence in the
base call at that position.  This parameter is only relevant
if you are using a base calling program that provides
quality information (for example phred).</p>


<h3><a name="SEQUENCE_FORCE_LEFT_START">SEQUENCE_FORCE_LEFT_START (int; default -1)</a></h3>

<p>Forces the 5' end of the left primer to be at the indicated 
position. Primers are also picked if they violate certain
constraints.</p>


<h3><a name="SEQUENCE_FORCE_LEFT_END">SEQUENCE_FORCE_LEFT_END (int; default -1)</a></h3>

<p>Forces the 3' end of the left primer to be at the indicated 
position. Primers are also picked if they violate certain
constraints.</p>


<h3><a name="SEQUENCE_FORCE_RIGHT_START">SEQUENCE_FORCE_RIGHT_START (int; default -1)</a></h3>

<p>Forces the 5' end of the right primer to be at the indicated 
position. Primers are also picked if they violate certain
constraints.</p>


<h3><a name="SEQUENCE_FORCE_RIGHT_END">SEQUENCE_FORCE_RIGHT_END (int; default -1)</a></h3>

<p>Forces the 3' end of the right primer to be at the indicated 
position. Primers are also picked if they violate certain
constraints.</p>


<h2><a name="globalTags">9. "GLOBAL" INPUT TAGS</a></h2>

<p>"Global" input tags start with PRIMER_... and describe the 
general parameters that primer3 should use in its searches. 
The values of these tags persist between input Boulder 
records until or unless they are explicitly reset. Errors 
in "Global" input tags are fatal because they invalidate 
the basic conditions under which primers are being picked.<br>
<br>
Because the laboratory detection step using internal oligos is 
independent of the PCR amplification procedure, internal oligo 
tags have defaults that are independent of the parameters that 
govern the selection of PCR primers. For example, the melting 
temperature of an oligo used for hybridization might be 
considerably lower than that used as a PCR primer.
<br></p>
<table style="text-align: left; width: 800px;" border="1">
	  <tr>
	    <td><a href="#PRIMER_DNA_CONC">PRIMER_DNA_CONC</a></td>
	    <td><a href="#PRIMER_LIBERAL_BASE">PRIMER_LIBERAL_BASE</a></td>
	    <td><a href="#PRIMER_PAIR_WT_PR_PENALTY">PRIMER_PAIR_WT_PR_PENALTY</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_DNTP_CONC">PRIMER_DNTP_CONC</a></td>
	    <td><a href="#PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS">PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS</a></td>
	    <td><a href="#PRIMER_PAIR_WT_TEMPLATE_MISPRIMING">PRIMER_PAIR_WT_TEMPLATE_MISPRIMING</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_EXPLAIN_FLAG">PRIMER_EXPLAIN_FLAG</a></td>
	    <td><a href="#PRIMER_LOWERCASE_MASKING">PRIMER_LOWERCASE_MASKING</a></td>
	    <td><a href="#PRIMER_PICK_ANYWAY">PRIMER_PICK_ANYWAY</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_FIRST_BASE_INDEX">PRIMER_FIRST_BASE_INDEX</a></td>
	    <td><a href="#PRIMER_MAX_END_GC">PRIMER_MAX_END_GC</a></td>
	    <td><a href="#PRIMER_PICK_INTERNAL_OLIGO">PRIMER_PICK_INTERNAL_OLIGO</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_GC_CLAMP">PRIMER_GC_CLAMP</a></td>
	    <td><a href="#PRIMER_MAX_END_STABILITY">PRIMER_MAX_END_STABILITY</a></td>
	    <td><a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INSIDE_PENALTY">PRIMER_INSIDE_PENALTY</a></td>
	    <td><a href="#PRIMER_MAX_GC">PRIMER_MAX_GC</a></td>
	    <td><a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_DNA_CONC">PRIMER_INTERNAL_DNA_CONC</a></td>
	    <td><a href="#PRIMER_MAX_LIBRARY_MISPRIMING">PRIMER_MAX_LIBRARY_MISPRIMING</a></td>
	    <td><a href="#PRIMER_POS_OVERLAP_TO_END_DIST">PRIMER_POS_OVERLAP_TO_END_DIST</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_DNTP_CONC">PRIMER_INTERNAL_DNTP_CONC</a></td>
	    <td><a href="#PRIMER_MAX_NS_ACCEPTED">PRIMER_MAX_NS_ACCEPTED</a></td>
	    <td><a href="#PRIMER_PRODUCT_MAX_TM">PRIMER_PRODUCT_MAX_TM</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_GC">PRIMER_INTERNAL_MAX_GC</a></td>
	    <td><a href="#PRIMER_MAX_POLY_X">PRIMER_MAX_POLY_X</a></td>
	    <td><a href="#PRIMER_PRODUCT_MIN_TM">PRIMER_PRODUCT_MIN_TM</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_LIBRARY_MISHYB">PRIMER_INTERNAL_MAX_LIBRARY_MISHYB</a></td>
	    <td><a href="#PRIMER_MAX_SELF_ANY">PRIMER_MAX_SELF_ANY</a></td>
	    <td><a href="#PRIMER_PRODUCT_OPT_SIZE">PRIMER_PRODUCT_OPT_SIZE</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_NS_ACCEPTED">PRIMER_INTERNAL_MAX_NS_ACCEPTED</a></td>
	    <td><a href="#PRIMER_MAX_SELF_END">PRIMER_MAX_SELF_END</a></td>
	    <td><a href="#PRIMER_PRODUCT_OPT_TM">PRIMER_PRODUCT_OPT_TM</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_POLY_X">PRIMER_INTERNAL_MAX_POLY_X</a></td>
	    <td><a href="#PRIMER_MAX_SIZE">PRIMER_MAX_SIZE</a></td>
	    <td><a href="#PRIMER_PRODUCT_SIZE_RANGE">PRIMER_PRODUCT_SIZE_RANGE</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_SELF_ANY">PRIMER_INTERNAL_MAX_SELF_ANY</a></td>
	    <td><a href="#PRIMER_MAX_TEMPLATE_MISPRIMING">PRIMER_MAX_TEMPLATE_MISPRIMING</a></td>
	    <td><a href="#PRIMER_QUALITY_RANGE_MAX">PRIMER_QUALITY_RANGE_MAX</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_SELF_END">PRIMER_INTERNAL_MAX_SELF_END</a></td>
	    <td><a href="#PRIMER_MAX_TM">PRIMER_MAX_TM</a></td>
	    <td><a href="#PRIMER_QUALITY_RANGE_MIN">PRIMER_QUALITY_RANGE_MIN</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_SIZE">PRIMER_INTERNAL_MAX_SIZE</a></td>
	    <td><a href="#PRIMER_MIN_END_QUALITY">PRIMER_MIN_END_QUALITY</a></td>
	    <td><a href="#PRIMER_SALT_CORRECTIONS">PRIMER_SALT_CORRECTIONS</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_TEMPLATE_MISHYB">PRIMER_INTERNAL_MAX_TEMPLATE_MISHYB</a></td>
	    <td><a href="#PRIMER_MIN_GC">PRIMER_MIN_GC</a></td>
	    <td><a href="#PRIMER_SALT_DIVALENT">PRIMER_SALT_DIVALENT</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MAX_TM">PRIMER_INTERNAL_MAX_TM</a></td>
	    <td><a href="#PRIMER_MIN_QUALITY">PRIMER_MIN_QUALITY</a></td>
	    <td><a href="#PRIMER_SALT_MONOVALENT">PRIMER_SALT_MONOVALENT</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MIN_GC">PRIMER_INTERNAL_MIN_GC</a></td>
	    <td><a href="#PRIMER_MIN_SIZE">PRIMER_MIN_SIZE</a></td>
	    <td><a href="#PRIMER_SEQUENCING_ACCURACY">PRIMER_SEQUENCING_ACCURACY</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MIN_QUALITY">PRIMER_INTERNAL_MIN_QUALITY</a></td>
	    <td><a href="#PRIMER_MIN_THREE_PRIME_DISTANCE">PRIMER_MIN_THREE_PRIME_DISTANCE</a></td>
	    <td><a href="#PRIMER_SEQUENCING_INTERVAL">PRIMER_SEQUENCING_INTERVAL</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MIN_SIZE">PRIMER_INTERNAL_MIN_SIZE</a></td>
	    <td><a href="#PRIMER_MIN_TM">PRIMER_MIN_TM</a></td>
	    <td><a href="#PRIMER_SEQUENCING_LEAD">PRIMER_SEQUENCING_LEAD</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MIN_TM">PRIMER_INTERNAL_MIN_TM</a></td>
	    <td><a href="#PRIMER_MISPRIMING_LIBRARY">PRIMER_MISPRIMING_LIBRARY</a></td>
	    <td><a href="#PRIMER_SEQUENCING_SPACING">PRIMER_SEQUENCING_SPACING</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_MISHYB_LIBRARY">PRIMER_INTERNAL_MISHYB_LIBRARY</a></td>
	    <td><a href="#PRIMER_NUM_RETURN">PRIMER_NUM_RETURN</a></td>
	    <td><a href="#PRIMER_TASK">PRIMER_TASK</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_OPT_GC_PERCENT">PRIMER_INTERNAL_OPT_GC_PERCENT</a></td>
	    <td><a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a></td>
	    <td><a href="#PRIMER_TM_FORMULA">PRIMER_TM_FORMULA</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_OPT_SIZE">PRIMER_INTERNAL_OPT_SIZE</a></td>
	    <td><a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a></td>
	    <td><a href="#PRIMER_WT_END_QUAL">PRIMER_WT_END_QUAL</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_OPT_TM">PRIMER_INTERNAL_OPT_TM</a></td>
	    <td><a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a></td>
	    <td><a href="#PRIMER_WT_END_STABILITY">PRIMER_WT_END_STABILITY</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_SALT_DIVALENT">PRIMER_INTERNAL_SALT_DIVALENT</a></td>
	    <td><a href="#PRIMER_OUTSIDE_PENALTY">PRIMER_OUTSIDE_PENALTY</a></td>
	    <td><a href="#PRIMER_WT_GC_PERCENT_GT">PRIMER_WT_GC_PERCENT_GT</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_SALT_MONOVALENT">PRIMER_INTERNAL_SALT_MONOVALENT</a></td>
	    <td><a href="#PRIMER_PAIR_MAX_COMPL_ANY">PRIMER_PAIR_MAX_COMPL_ANY</a></td>
	    <td><a href="#PRIMER_WT_GC_PERCENT_LT">PRIMER_WT_GC_PERCENT_LT</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_END_QUAL">PRIMER_INTERNAL_WT_END_QUAL</a></td>
	    <td><a href="#PRIMER_PAIR_MAX_COMPL_END">PRIMER_PAIR_MAX_COMPL_END</a></td>
	    <td><a href="#PRIMER_WT_LIBRARY_MISPRIMING">PRIMER_WT_LIBRARY_MISPRIMING</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_GC_PERCENT_GT">PRIMER_INTERNAL_WT_GC_PERCENT_GT</a></td>
	    <td><a href="#PRIMER_PAIR_MAX_DIFF_TM">PRIMER_PAIR_MAX_DIFF_TM</a></td>
	    <td><a href="#PRIMER_WT_NUM_NS">PRIMER_WT_NUM_NS</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_GC_PERCENT_LT">PRIMER_INTERNAL_WT_GC_PERCENT_LT</a></td>
	    <td><a href="#PRIMER_PAIR_MAX_LIBRARY_MISPRIMING">PRIMER_PAIR_MAX_LIBRARY_MISPRIMING</a></td>
	    <td><a href="#PRIMER_WT_POS_PENALTY">PRIMER_WT_POS_PENALTY</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_LIBRARY_MISHYB">PRIMER_INTERNAL_WT_LIBRARY_MISHYB</a></td>
	    <td><a href="#PRIMER_PAIR_MAX_TEMPLATE_MISPRIMING">PRIMER_PAIR_MAX_TEMPLATE_MISPRIMING</a></td>
	    <td><a href="#PRIMER_WT_SELF_ANY">PRIMER_WT_SELF_ANY</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_NUM_NS">PRIMER_INTERNAL_WT_NUM_NS</a></td>
	    <td><a href="#PRIMER_PAIR_WT_COMPL_ANY">PRIMER_PAIR_WT_COMPL_ANY</a></td>
	    <td><a href="#PRIMER_WT_SELF_END">PRIMER_WT_SELF_END</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_SELF_ANY">PRIMER_INTERNAL_WT_SELF_ANY</a></td>
	    <td><a href="#PRIMER_PAIR_WT_COMPL_END">PRIMER_PAIR_WT_COMPL_END</a></td>
	    <td><a href="#PRIMER_WT_SEQ_QUAL">PRIMER_WT_SEQ_QUAL</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_SELF_END">PRIMER_INTERNAL_WT_SELF_END</a></td>
	    <td><a href="#PRIMER_PAIR_WT_DIFF_TM">PRIMER_PAIR_WT_DIFF_TM</a></td>
	    <td><a href="#PRIMER_WT_SIZE_GT">PRIMER_WT_SIZE_GT</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_SEQ_QUAL">PRIMER_INTERNAL_WT_SEQ_QUAL</a></td>
	    <td><a href="#PRIMER_PAIR_WT_IO_PENALTY">PRIMER_PAIR_WT_IO_PENALTY</a></td>
	    <td><a href="#PRIMER_WT_SIZE_LT">PRIMER_WT_SIZE_LT</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_SIZE_GT">PRIMER_INTERNAL_WT_SIZE_GT</a></td>
	    <td><a href="#PRIMER_PAIR_WT_LIBRARY_MISPRIMING">PRIMER_PAIR_WT_LIBRARY_MISPRIMING</a></td>
	    <td><a href="#PRIMER_WT_TEMPLATE_MISPRIMING">PRIMER_WT_TEMPLATE_MISPRIMING</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_SIZE_LT">PRIMER_INTERNAL_WT_SIZE_LT</a></td>
	    <td><a href="#PRIMER_PAIR_WT_PRODUCT_SIZE_GT">PRIMER_PAIR_WT_PRODUCT_SIZE_GT</a></td>
	    <td><a href="#PRIMER_WT_TM_GT">PRIMER_WT_TM_GT</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_TEMPLATE_MISHYB">PRIMER_INTERNAL_WT_TEMPLATE_MISHYB</a></td>
	    <td><a href="#PRIMER_PAIR_WT_PRODUCT_SIZE_LT">PRIMER_PAIR_WT_PRODUCT_SIZE_LT</a></td>
	    <td><a href="#PRIMER_WT_TM_LT">PRIMER_WT_TM_LT</a></td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_TM_GT">PRIMER_INTERNAL_WT_TM_GT</a></td>
	    <td><a href="#PRIMER_PAIR_WT_PRODUCT_TM_GT">PRIMER_PAIR_WT_PRODUCT_TM_GT</a></td>
	    <td>&nbsp;</td>
	  </tr>
	  <tr>
	    <td><a href="#PRIMER_INTERNAL_WT_TM_LT">PRIMER_INTERNAL_WT_TM_LT</a></td>
	    <td><a href="#PRIMER_PAIR_WT_PRODUCT_TM_LT">PRIMER_PAIR_WT_PRODUCT_TM_LT</a></td>
	    <td>&nbsp;</td>
	  </tr>
</table>
<h3><a name="PRIMER_TASK">PRIMER_TASK (string; default pick_pcr_primers)</a></h3>

<p>This tag tells primer3 what task to perform.
Legal values are:<br>
<br>
   <tt><i>pick_detection_primers</i></tt><br>
<br>
   Pick primers to detect a sequence (the classic primer3 task). Use 
   the <a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a>, <a href="#PRIMER_PICK_INTERNAL_OLIGO">PRIMER_PICK_INTERNAL_OLIGO</a> and 
   <a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a> to control which primers are picked.
   If <a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a> and <a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a> are 
   selected primer3 tries to pick primer pairs.<br>
<br>
   <tt><i>check_primers</i></tt><br>
<br>
   Primer3 does not pick any primers, it only checks the primers 
   provided in <a href="#SEQUENCE_PRIMER">SEQUENCE_PRIMER</a>, <a href="#SEQUENCE_INTERNAL_OLIGO">SEQUENCE_INTERNAL_OLIGO</a> and 
   <a href="#SEQUENCE_PRIMER_REVCOMP">SEQUENCE_PRIMER_REVCOMP</a>. Its the only task which does not
   require a sequence. If the sequence is provided, it is used
   as a template.<br>
<br>
   <tt><i>pick_primer_list</i></tt><br>
<br>
   Pick all primers in the sequence or limited by included region.
   It returns the primers sorted by quality starting with the 
   best primers. If <a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a> and 
   <a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a> is selected primer3 does not to pick 
   primer pairs but treats it as independent lists.<br>
<br>
   <tt><i>pick_sequencing_primers</i></tt><br>
<br>
   Pick primers suited to sequence a region. <a href="#SEQUENCE_TARGET">SEQUENCE_TARGET</a> can be 
   used to indicate several targets. The position of each primer is 
   calculated for optimal sequencing results.<br>
<br>
   <tt><i>pick_cloning_primers</i></tt><br>
<br>
   Pick primers suited to clone a gene were the start nucleotide and 
   the end nucleotide of the PCR fragment must be fixed, for example 
   to clone an ORF. <a href="#SEQUENCE_INCLUDED_REGION">SEQUENCE_INCLUDED_REGION</a> must be used to 
   indicate the first and the last nucleotide. Due to these 
   limitations primer3 can only vary the length of the primers. Set 
   <a href="#PRIMER_PICK_ANYWAY">PRIMER_PICK_ANYWAY</a>=1 to obtain primers even if they violate 
   specific constraints.<br>
<br>
   <tt><i>pick_discriminative_primers</i></tt><br>
<br>
   Pick primers suited to select primers which bind with their end at 
   a specific position. This can be used to force the end of a primer 
   to a polymorphic site, with the goal of discriminating between
   sequence variants. <a href="#SEQUENCE_INCLUDED_REGION">SEQUENCE_INCLUDED_REGION</a> must be used to 
   indicate the last nucleotide of the left (first nucleotide of 
   included region) and the right primer (last nucleotide of 
   included region). Due to these limitations primer3 can only vary 
   the length of the primers. Set <a href="#PRIMER_PICK_ANYWAY">PRIMER_PICK_ANYWAY</a>=1 to obtain 
   primers even if they violate specific constraints.<br>
<br>
   <tt><i>pick_pcr_primers</i></tt><br>
<br>
   Shortcut for the following settings:<br>
   <a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_detection_primers<br>
   <a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a>=1<br>
   <a href="#PRIMER_PICK_INTERNAL_OLIGO">PRIMER_PICK_INTERNAL_OLIGO</a>=0<br>
   <a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a>=1<br>
<br>
   <tt><i>pick_pcr_primers_and_hyb_probe</i></tt><br>
<br>
   Shortcut for the following settings:<br>
   <a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_detection_primers<br>
   <a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a>=1<br>
   <a href="#PRIMER_PICK_INTERNAL_OLIGO">PRIMER_PICK_INTERNAL_OLIGO</a>=1<br>
   <a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a>=1<br>
<br>
   <tt><i>pick_left_only</i></tt><br>
<br>
   Shortcut for the following settings:<br>
   <a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_detection_primers<br>
   <a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a>=1<br>
   <a href="#PRIMER_PICK_INTERNAL_OLIGO">PRIMER_PICK_INTERNAL_OLIGO</a>=0<br>
   <a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a>=0<br>
<br>
   <tt><i>pick_right_only</i></tt><br>
<br>
   Shortcut for the following settings:<br>
   <a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_detection_primers<br>
   <a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a>=0<br>
   <a href="#PRIMER_PICK_INTERNAL_OLIGO">PRIMER_PICK_INTERNAL_OLIGO</a>=0<br>
   <a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a>=1<br>
<br>
   <tt><i>pick_hyb_probe_only</i></tt><br>
<br>
   Shortcut for the following settings:<br>
   <a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_detection_primers<br>
   <a href="#PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER</a>=0<br>
   <a href="#PRIMER_PICK_INTERNAL_OLIGO">PRIMER_PICK_INTERNAL_OLIGO</a>=1<br>
   <a href="#PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER</a>=0<br>
</p>


<h3><a name="PRIMER_PICK_LEFT_PRIMER">PRIMER_PICK_LEFT_PRIMER (boolean; default 1)</a></h3>

<p>If the associated value is 1 (non-0), then primer3 will attempt to
pick left primers.</p>


<h3><a name="PRIMER_PICK_INTERNAL_OLIGO">PRIMER_PICK_INTERNAL_OLIGO (boolean; default 0)</a></h3>

<p>If the associated value is 1 (non-0), then primer3 will attempt to
pick an internal oligo (hybridization probe to detect the PCR
product).</p>


<h3><a name="PRIMER_PICK_RIGHT_PRIMER">PRIMER_PICK_RIGHT_PRIMER (boolean; default 1)</a></h3>

<p>If the associated value is 1 (non-0), then primer3 will attempt to
pick a right primer.</p>


<h3><a name="PRIMER_NUM_RETURN">PRIMER_NUM_RETURN (int; default 5)</a></h3>

<p>The maximum number of primer (pairs) to return.  Primer pairs
returned are sorted by their &quot;quality&quot;, in other words by the
value of the objective function (where a lower number indicates a
better primer pair).  Caution: setting this parameter to a large
value will increase running time.</p>


<h3><a name="PRIMER_POS_OVERLAP_TO_END_DIST">PRIMER_POS_OVERLAP_TO_END_DIST (int; default 5)</a></h3>

<p>The minimum distance of nucleotides a end of a primer has to 
have to one of the positions in <a href="#SEQUENCE_PRIMER_OVERLAP_POS">SEQUENCE_PRIMER_OVERLAP_POS</a>.</p>


<h3><a name="PRIMER_PRODUCT_SIZE_RANGE">PRIMER_PRODUCT_SIZE_RANGE (size range list; default 100-300)</a></h3>

<p>The associated values specify the lengths of the product that the
user wants the primers to create, and is a space separated list
of elements of the form</p>
<pre>
<tt><i>&lt;x&gt;-&lt;y&gt;</i></tt>
</pre>
<p>where an <tt><i>&lt;x&gt;-&lt;y&gt;</i></tt> pair is a legal range of lengths for the
product.  For example, if one wants PCR products to be between
100 to 150 bases (inclusive) then one would set this parameter to
<tt><i>100-150</i></tt>.  If one desires PCR products in either the range from
100 to 150 bases or in the range from 200 to 250 bases then one
would set this parameter to <tt><i>100-150 200-250</i></tt>.<br>
<br>
Primer3 favors product-size ranges to the left side of the parameter string.
Primer3 will return legal primers pairs in the first range
regardless the value of the objective function for these pairs.
Only if there are an insufficient number of primers in the first
range will primer3 return primers in a subsequent range.<br>
<br>
For those with primarily a computational background,
the PCR product size is the size (in base pairs) 
of the DNA fragment that would be produced by the
PCR reaction on the given sequence template.  This
would, of course, include the primers themselves.</p>


<h3><a name="PRIMER_PRODUCT_OPT_SIZE">PRIMER_PRODUCT_OPT_SIZE (int; default 0)</a></h3>

<p>The optimum size for the PCR product.  0 indicates that there is
no optimum product size.  This parameter influences primer pair 
selection only if <a href="#PRIMER_PAIR_WT_PRODUCT_SIZE_GT">PRIMER_PAIR_WT_PRODUCT_SIZE_GT</a> or
<a href="#PRIMER_PAIR_WT_PRODUCT_SIZE_LT">PRIMER_PAIR_WT_PRODUCT_SIZE_LT</a> is non-0.<br>
This parameter increases calculation time, so set this only if 
you need a exact product size.</p>


<h3><a name="PRIMER_PAIR_WT_PRODUCT_SIZE_LT">PRIMER_PAIR_WT_PRODUCT_SIZE_LT (float; default 0.0)</a></h3>

<p>Penalty weight for products shorter than <a href="#PRIMER_PRODUCT_OPT_SIZE">PRIMER_PRODUCT_OPT_SIZE</a>.</p>


<h3><a name="PRIMER_PAIR_WT_PRODUCT_SIZE_GT">PRIMER_PAIR_WT_PRODUCT_SIZE_GT (float; default 0.0)</a></h3>

<p>Penalty weight for products longer than <a href="#PRIMER_PRODUCT_OPT_SIZE">PRIMER_PRODUCT_OPT_SIZE</a>.</p>


<h3><a name="PRIMER_MIN_SIZE">PRIMER_MIN_SIZE (int; default 18)</a></h3>

<p>Minimum acceptable length of a primer.  Must be greater than 0
and less than or equal to <a href="#PRIMER_MAX_SIZE">PRIMER_MAX_SIZE</a>.</p>


<h3><a name="PRIMER_INTERNAL_MIN_SIZE">PRIMER_INTERNAL_MIN_SIZE (int; default 18)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MIN_SIZE">PRIMER_MIN_SIZE</a> for the internal oligo.</p>


<h3><a name="PRIMER_OPT_SIZE">PRIMER_OPT_SIZE (int; default 20)</a></h3>

<p>Optimum length (in bases) of a primer. Primer3 will attempt to 
pick primers close to this length.</p>


<h3><a name="PRIMER_INTERNAL_OPT_SIZE">PRIMER_INTERNAL_OPT_SIZE (int; default 20)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> for the internal oligo.</p>


<h3><a name="PRIMER_MAX_SIZE">PRIMER_MAX_SIZE (int; default 27)</a></h3>

<p>Maximum acceptable length (in bases) of a primer.  Currently this
parameter cannot be larger than 35.  This limit is governed by
maximum oligo size for which primer3's melting-temperature is
valid.</p>


<h3><a name="PRIMER_INTERNAL_MAX_SIZE">PRIMER_INTERNAL_MAX_SIZE (int; default 27)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MAX_SIZE">PRIMER_MAX_SIZE</a> for the internal oligo.</p>


<h3><a name="PRIMER_WT_SIZE_LT">PRIMER_WT_SIZE_LT (float; default 1.0)</a></h3>

<p>Penalty weight for primers shorter than <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_SIZE_LT">PRIMER_INTERNAL_WT_SIZE_LT (float; default 1.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_SIZE_LT">PRIMER_WT_SIZE_LT</a> for the internal oligo.</p>


<h3><a name="PRIMER_WT_SIZE_GT">PRIMER_WT_SIZE_GT (float; default 1.0)</a></h3>

<p>Penalty weight for primers longer than <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_SIZE_GT">PRIMER_INTERNAL_WT_SIZE_GT (float; default 1.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_INTERNAL_WT_SIZE_GT">PRIMER_INTERNAL_WT_SIZE_GT</a> for the internal oligo.</p>


<h3><a name="PRIMER_MIN_GC">PRIMER_MIN_GC (float; default 20.0)</a></h3>

<p>Minimum allowable percentage of Gs and Cs in any primer.</p>


<h3><a name="PRIMER_INTERNAL_MIN_GC">PRIMER_INTERNAL_MIN_GC (float; default 20.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MIN_GC">PRIMER_MIN_GC</a> for the internal oligo.</p>


<h3><a name="PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT (float; default 50.0)</a></h3>

<p>Optimum GC percent.  This parameter influences primer selection only if
<a href="#PRIMER_WT_GC_PERCENT_GT">PRIMER_WT_GC_PERCENT_GT</a> or <a href="#PRIMER_WT_GC_PERCENT_LT">PRIMER_WT_GC_PERCENT_LT</a> are non-0.</p>


<h3><a name="PRIMER_INTERNAL_OPT_GC_PERCENT">PRIMER_INTERNAL_OPT_GC_PERCENT (float; default 50.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> for the internal oligo.</p>


<h3><a name="PRIMER_MAX_GC">PRIMER_MAX_GC (float; default 80.0)</a></h3>

<p>Maximum allowable percentage of Gs and Cs in any primer generated
by Primer.</p>


<h3><a name="PRIMER_INTERNAL_MAX_GC">PRIMER_INTERNAL_MAX_GC (float; default 80.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MAX_GC">PRIMER_MAX_GC</a> for the internal oligo.</p>


<h3><a name="PRIMER_WT_GC_PERCENT_LT">PRIMER_WT_GC_PERCENT_LT (float; default 0.0)</a></h3>

<p>Penalty weight for primers with GC percent lower than
<a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_GC_PERCENT_LT">PRIMER_INTERNAL_WT_GC_PERCENT_LT (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_GC_PERCENT_LT">PRIMER_WT_GC_PERCENT_LT</a> for the internal oligo.</p>


<h3><a name="PRIMER_WT_GC_PERCENT_GT">PRIMER_WT_GC_PERCENT_GT (float; default 0.0)</a></h3>

<p>Penalty weight for primers with GC percent higher than
<a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_GC_PERCENT_GT">PRIMER_INTERNAL_WT_GC_PERCENT_GT (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_GC_PERCENT_GT">PRIMER_WT_GC_PERCENT_GT</a> for the internal oligo.</p>


<h3><a name="PRIMER_GC_CLAMP">PRIMER_GC_CLAMP (int; default 0)</a></h3>

<p>Require the specified number of consecutive Gs and Cs at the 3'
end of both the left and right primer.  (This parameter has no
effect on the internal oligo if one is requested.)</p>


<h3><a name="PRIMER_MAX_END_GC">PRIMER_MAX_END_GC (int; default 5)</a></h3>

<p>The maximum number of Gs or Cs allowed in the last five 3' 
bases of a left or right primer.</p>


<h3><a name="PRIMER_MIN_TM">PRIMER_MIN_TM (float; default 57.0)</a></h3>

<p>Minimum acceptable melting temperature (Celsius) for a primer
oligo.</p>


<h3><a name="PRIMER_INTERNAL_MIN_TM">PRIMER_INTERNAL_MIN_TM (float; default 57.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MIN_TM">PRIMER_MIN_TM</a> for the internal oligo.</p>


<h3><a name="PRIMER_OPT_TM">PRIMER_OPT_TM (float; default 60.0)</a></h3>

<p>Optimum melting temperature (Celsius) for a primer. Primer3
will try to pick primers with melting temperatures are close to
this temperature.  The oligo melting temperature formula used can
be specified by user. Please see <a href="#PRIMER_TM_FORMULA">PRIMER_TM_FORMULA</a> for more
information.</p>


<h3><a name="PRIMER_INTERNAL_OPT_TM">PRIMER_INTERNAL_OPT_TM (float; default 60.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> for the internal oligo.</p>


<h3><a name="PRIMER_MAX_TM">PRIMER_MAX_TM (float; default 63.0)</a></h3>

<p>Maximum acceptable melting temperature (Celsius) for a primer
oligo.</p>


<h3><a name="PRIMER_INTERNAL_MAX_TM">PRIMER_INTERNAL_MAX_TM (float; default 63.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MAX_TM">PRIMER_MAX_TM</a> for the internal oligo.</p>


<h3><a name="PRIMER_PAIR_MAX_DIFF_TM">PRIMER_PAIR_MAX_DIFF_TM (float; default 100.0)</a></h3>

<p>Maximum acceptable (unsigned) difference between the melting
temperatures of the left and right primers.</p>


<h3><a name="PRIMER_WT_TM_LT">PRIMER_WT_TM_LT (float; default 1.0)</a></h3>

<p>Penalty weight for primers with Tm lower than <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_TM_LT">PRIMER_INTERNAL_WT_TM_LT (float; default 1.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_TM_LT">PRIMER_WT_TM_LT</a> for the internal oligo.</p>


<h3><a name="PRIMER_WT_TM_GT">PRIMER_WT_TM_GT (float; default 1.0)</a></h3>

<p>Penalty weight for primers with Tm over <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_TM_GT">PRIMER_INTERNAL_WT_TM_GT (float; default 1.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_TM_GT">PRIMER_WT_TM_GT</a> for the internal oligo.</p>


<h3><a name="PRIMER_PAIR_WT_DIFF_TM">PRIMER_PAIR_WT_DIFF_TM (float; default 0.0)</a></h3>

<p>Penalty weight for the TM difference between the left primer and 
the right primer.</p>


<h3><a name="PRIMER_PRODUCT_MIN_TM">PRIMER_PRODUCT_MIN_TM (float; default -1000000.0)</a></h3>

<p>The minimum allowed melting temperature of the amplicon.  Please
see the documentation on <a href="#PRIMER_PRODUCT_MAX_TM">PRIMER_PRODUCT_MAX_TM</a> for details.</p>


<h3><a name="PRIMER_PRODUCT_OPT_TM">PRIMER_PRODUCT_OPT_TM (float; default 0.0)</a></h3>

<p>The optimum melting temperature for the PCR product. 0 indicates
that there is no optimum temperature.</p>


<h3><a name="PRIMER_PRODUCT_MAX_TM">PRIMER_PRODUCT_MAX_TM (float; default 1000000.0)</a></h3>

<p>The maximum allowed melting temperature of the amplicon.  Primer3
calculates product Tm calculated using the formula from Bolton
and McCarthy, PNAS 84:1390 (1962) as presented in Sambrook,
Fritsch and Maniatis, Molecular Cloning, p 11.46 (1989, CSHL
Press).</p>
<pre>
   Tm = 81.5 + 16.6(log10([Na+])) + .41*(%GC) - 600/length
</pre>
<p>Where [Na+] is the molar sodium concentration, (%GC) is the
percent of Gs and Cs in the sequence, and length is the length of
the sequence.<br>
<br>
A similar formula is used by the prime primer selection program
in GCG (<a href="http://www.gcg.com">http://www.gcg.com</a>), which instead uses 675.0 / length in
the last term (after F. Baldino, Jr, M.-F. Chesselet, and M.E.
Lewis, Methods in Enzymology 168:766 (1989) eqn (1) on page 766
without the mismatch and formamide terms).  The formulas here and
in Baldino et al. assume Na+ rather than K+.  According to
J.G. Wetmur, Critical Reviews in BioChem. and Mol. Bio. 26:227
(1991) 50 mM K+ should be equivalent in these formulae to .2 M
Na+.  Primer3 uses the same salt concentration value for
calculating both the primer melting temperature and the oligo
melting temperature.  If you are planning to use the PCR product
for hybridization later this behavior will not give you the Tm
under hybridization conditions.</p>


<h3><a name="PRIMER_PAIR_WT_PRODUCT_TM_LT">PRIMER_PAIR_WT_PRODUCT_TM_LT (float; default 0.0)</a></h3>

<p>Penalty weight for products with a Tm lower than 
<a href="#PRIMER_PRODUCT_OPT_TM">PRIMER_PRODUCT_OPT_TM</a>.</p>


<h3><a name="PRIMER_PAIR_WT_PRODUCT_TM_GT">PRIMER_PAIR_WT_PRODUCT_TM_GT (float; default 0.0)</a></h3>

<p>Penalty weight for products with a Tm higher than  
<a href="#PRIMER_PRODUCT_OPT_TM">PRIMER_PRODUCT_OPT_TM</a>.</p>


<h3><a name="PRIMER_TM_FORMULA">PRIMER_TM_FORMULA (int; default 0)</a></h3>

<p>Specifies details of melting temperature calculation.  (New in
v. 1.1.0, added by Maido Remm and Triinu Koressaar.)<br>
<br>
A value of 0 directs primer3 to a backward compatible calculation
(in other words, the only calculation available in previous
version of primer3).<br>
<br>
This backward compatible calculation uses the table of
thermodynamic parameters in the paper [Breslauer KJ, Frank R,
Blöcker H and Marky LA (1986) "Predicting DNA duplex stability
from the base sequence" Proc Natl Acad Sci 83:4746-50
<a href="http://dx.doi.org/10.1073/pnas.83.11.3746">http://dx.doi.org/10.1073/pnas.83.11.3746</a>],
and the method in the paper [Rychlik W, Spencer WJ and Rhoads
RE (1990) "Optimization of the annealing temperature for DNA
amplification in vitro", Nucleic Acids Res 18:6409-12
<a href="http://dx.doi.org/10.1093/nar/18.21.6409">http://dx.doi.org/10.1093/nar/18.21.6409</a>].<br>
<br>
A value of 1 (*RECOMMENDED*) directs primer3 to use the table of
thermodynamic values and the method for melting temperature
calculation suggested in the paper [SantaLucia JR (1998) "A unified
view of polymer, dumbbell and oligonucleotide DNA nearest-neighbor
thermodynamics", Proc Natl Acad Sci 95:1460-65
<a href="http://dx.doi.org/10.1073/pnas.95.4.1460">http://dx.doi.org/10.1073/pnas.95.4.1460</a>].<br>
<br>
Use tag <a href="#PRIMER_SALT_CORRECTIONS">PRIMER_SALT_CORRECTIONS</a>, to specify the salt correction
method for melting temperature calculation.<br>
<br>
Example of calculating the melting temperature of an oligo if
<a href="#PRIMER_TM_SANTALUCIA">PRIMER_TM_SANTALUCIA</a>=1 and <a href="#PRIMER_SALT_CORRECTIONS">PRIMER_SALT_CORRECTIONS</a>=1
recommended values):<br>
<br>
primer=CGTGACGTGACGGACT<br>
<br>
Using default salt and DNA concentrations we have</p>
<pre>
Tm = deltaH/(deltaS + R*ln(C/4))
</pre>
<p>where R is the gas constant (1.987 cal/K mol) and C is the DNA 
concentration.</p>
<pre>
deltaH(predicted) =
  = dH(CG) + dH(GT) + dH(TG) + .. + dH(CT) +
     + dH(init.w.term.GC) + dH(init.w.term.AT) =

  = -10.6 + (-8.4) + (-8.5) + .. + (-7.8) + 0.1 + 2.3  =

  = -128.8 kcal/mol
</pre>
<p>where 'init.w.term GC' and 'init.w.term AT' are two
initiation parameters for duplex formation: 'initiation with
terminal GC' and 'initiation with terminal AT'</p>
<pre>
deltaS(predicted) =
  = dS(CG) + dS(GT) + dS(TG) + .. + dS(CT) +
    + dS(init.w.term.GC) + dS(init.w.term.AT) =

  = -27.2 + (-22.4) + (-22.7) + .. + (-21.0) + (-2.8) + 4.1 =
 
  = -345.2 cal/k*mol

deltaS(salt corrected) = 
  = deltaS(predicted) + 0.368*15(NN pairs)*ln(0.05M monovalent cations) =
  = -361.736

Tm = -128.800/(-361.736+1.987*ln((5*10^(-8))/4)) =
   = 323.704 K

Tm(C) = 323.704 - 273.15 = 50.554 C
</pre>


<h3><a name="PRIMER_SALT_MONOVALENT">PRIMER_SALT_MONOVALENT (float; default 50.0)</a></h3>

<p>The millimolar (mM) concentration of monovalent salt cations (usually KCl) in the PCR.
Primer3 uses this argument to calculate oligo and primer melting
temperatures. Use tag <a href="#PRIMER_SALT_DIVALENT">PRIMER_SALT_DIVALENT</a> to specify the concentration
of divalent cations (in this case you also should use tag <a href="#PRIMER_DNTP_CONC">PRIMER_DNTP_CONC</a>).</p>


<h3><a name="PRIMER_INTERNAL_SALT_MONOVALENT">PRIMER_INTERNAL_SALT_MONOVALENT (float; default 50.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_SALT_MONOVALENT">PRIMER_SALT_MONOVALENT</a> for the internal oligo.</p>


<h3><a name="PRIMER_SALT_DIVALENT">PRIMER_SALT_DIVALENT (float; default 0.0)</a></h3>

<p>The millimolar concentration of divalent salt cations (usually MgCl^(2+)) in
the PCR. (New in v. 1.1.0, added by Maido Remm and Triinu Koressaar)<br>
<br>
Primer3 converts concentration of divalent cations to concentration
of monovalent cations using formula suggested in the paper [Ahsen von N,
Wittwer CT, Schutz E (2001) "Oligonucleotide Melting Temperatures under PCR
Conditions: Nearest-Neighbor Corrections for Mg^(2+), Deoxynucleotide Triphosphate,
and Dimethyl Sulfoxide Concentrations with Comparison to Alternative Empirical 
Formulas", Clinical Chemistry 47:1956-61 <a href="http://www.clinchem.org/cgi/content/full/47/11/1956">http://www.clinchem.org/cgi/content/full/47/11/1956</a>].</p>
<pre>
[Monovalent cations] = [Monovalent cations] + 120*(([divalent cations] - [dNTP])^0.5)
</pre>
<p>According to the formula concentration of deoxynucleotide triphosphate
[dNTP] must be smaller than concentration of divalent cations. If the
specified concentration of dNTPs is larger than specified concentration of
divalent cations then the effect of divalent cations is not considered. The
concentration of dNTPs is included to the formula because of some magnesium is bound by the
dNTP. Attained concentration of monovalent cations is used to calculate oligo/primer
melting temperature. Use tag <a href="#PRIMER_DNTP_CONC">PRIMER_DNTP_CONC</a> to specify the concentration of dNTPs.</p>


<h3><a name="PRIMER_INTERNAL_SALT_DIVALENT">PRIMER_INTERNAL_SALT_DIVALENT (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_SALT_DIVALENT">PRIMER_SALT_DIVALENT</a> for the internal oligo.</p>


<h3><a name="PRIMER_DNTP_CONC">PRIMER_DNTP_CONC (float; default 0.0)</a></h3>

<p>The millimolar concentration of the sum of all 
deoxyribonucleotide triphosphates. A reaction mix containing 
0.2 mM ATP, 0.2 mM CTP, 0.2 mM GTP and 0.2 mM TTP would have a 
PRIMER_DNTP_CONC=0.8. This argument is considered only if 
<a href="#PRIMER_SALT_DIVALENT">PRIMER_SALT_DIVALENT</a> is specified. See <a href="#PRIMER_SALT_DIVALENT">PRIMER_SALT_DIVALENT</a>.</p>


<h3><a name="PRIMER_INTERNAL_DNTP_CONC">PRIMER_INTERNAL_DNTP_CONC (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_DNTP_CONC">PRIMER_DNTP_CONC</a> for the internal oligo.</p>


<h3><a name="PRIMER_SALT_CORRECTIONS">PRIMER_SALT_CORRECTIONS (int; default 0)</a></h3>

<p>Specifies the salt correction formula for the melting temperature
calculation.  (New in v. 1.1.0, added by Maido Remm and Triinu
Koressaar)<br>
<br>
A value of 0 directs primer3 to use the the salt correction
formula in the paper [Schildkraut, C, and Lifson, S (1965)
"Dependence of the melting temperature of DNA on salt
concentration", Biopolymers 3:195-208 (not available on-line)].
This was the formula used in previous version of primer3.<br>
<br>
A value of 1 (*RECOMMENDED*) directs primer3 to use the salt correction
formula in the paper [SantaLucia JR (1998) "A unified view of polymer,
dumbbell and oligonucleotide DNA nearest-neighbor thermodynamics",
Proc Natl Acad Sci 95:1460-65
<a href="http://dx.doi.org/10.1073/pnas.95.4.1460">http://dx.doi.org/10.1073/pnas.95.4.1460</a>]<br>
<br>
A value of 2 directs primer3 to use the salt correction formula
in the paper [Owczarzy R, You Y, Moreira BG, Manthey JA, Huang L,
Behlke MA and Walder JA (2004) "Effects of sodium ions on DNA
duplex oligomers: Improved predictions of melting temperatures",
Biochemistry 43:3537-54 <a href="http://dx.doi.org/10.1021/bi034621r">http://dx.doi.org/10.1021/bi034621r</a>].</p>


<h3><a name="PRIMER_DNA_CONC">PRIMER_DNA_CONC (float; default 50.0)</a></h3>

<p>
<p>
A value to use as nanomolar (nM) concentration of each annealing oligo 
over the course the PCR.
Primer3 uses this argument to esimate oligo melting
temperatures. This parameter corresponds to 'c' in equation (ii) 
of the paper [SantaLucia (1998) A unified view of polymer, dumbbell, 
and oligonucleotide DNA nearest-neighbor thermodynamics. 
Proc Natl Acad Sci 95:1460-1465
<a href="http://www.pnas.org/content/95/4/1460.full.pdf+html">http://www.pnas.org/content/95/4/1460.full.pdf+html</a>],
where a suitable value (for a lower initial concentration of template) 
is "empirically determined".<br>
<br>
The default (50nM) works well with the standard
protocol used at the Whitehead/MIT Center for Genome 
Research--0.5 microliters of 20 micromolar concentration for each
primer in a 20 microliter reaction with 10 nanograms
template, 0.025 units/microliter Taq polymerase in 0.1 mM each
dNTP, 1.5mM MgCl2, 50mM KCl, 10mM Tris-HCL (pH 9.3) using 35
cycles with an annealing temperature of 56 degrees Celsius.<br>
<br>
The value of this parameter is less than the actual
concentration of oligos in the initial reaction mix because it is the
concentration of annealing oligos, which in turn depends on the
amount of template (including PCR product) in a given cycle.
This concentration increases a great deal during a PCR;
fortunately PCR seems quite robust for a variety of oligo melting
temperatures.
<br>
See ADVICE FOR PICKING PRIMERS.</p>


<h3><a name="PRIMER_INTERNAL_DNA_CONC">PRIMER_INTERNAL_DNA_CONC (float; default 50.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_DNA_CONC">PRIMER_DNA_CONC</a> for the internal oligo.</p>


<h3><a name="PRIMER_MAX_SELF_ANY">PRIMER_MAX_SELF_ANY (decimal, 9999.99; default 8.00)</a></h3>

<p>PRIMER_MAX_SELF_ANY describes the tendency of a primer to bind to 
itself (interfering with target sequence binding). It will score 
ANY binding occurring within the entire primer sequence.<br>
It is the maximum allowable local alignment score when testing 
a single primer for (local) self-complementarity and the 
maximum allowable local alignment score when testing for 
complementarity between left and right primers.  Local 
self-complementarity is taken to predict the tendency of primers 
to anneal to each other without necessarily causing self-priming 
in the PCR.  The scoring system gives 1.00 for complementary 
bases, -0.25 for a match of any base (or N) with an N, -1.00 for 
a mismatch, and -2.00 for a gap. Only single-base-pair gaps are 
allowed. For example, the alignment</p>
<pre>
   5' ATCGNA 3'
      || | |
   3' TA-CGT 5'
</pre>
is allowed (and yields a score of 1.75), but the alignment
<pre>
   5' ATCCGNA 3'
      ||  | |
   3' TA--CGT 5'
</pre>
<p>is not considered.  Scores are non-negative, and a score of 0.00
indicates that there is no reasonable local alignment between two
oligos.</p>


<h3><a name="PRIMER_INTERNAL_MAX_SELF_ANY">PRIMER_INTERNAL_MAX_SELF_ANY (decimal, 9999.99; default 12.00)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MAX_SELF_ANY">PRIMER_MAX_SELF_ANY</a> for the internal oligo.</p>


<h3><a name="PRIMER_PAIR_MAX_COMPL_ANY">PRIMER_PAIR_MAX_COMPL_ANY (decimal, 9999.99; default 8.00)</a></h3>

<p>PRIMER_PAIR_MAX_COMPL_ANY describes the tendency of the left primer 
to bind to the right primer. It is similar to <a href="#PRIMER_MAX_SELF_ANY">PRIMER_MAX_SELF_ANY</a>.</p>


<h3><a name="PRIMER_WT_SELF_ANY">PRIMER_WT_SELF_ANY (float; default 0.0)</a></h3>

<p>Penalty weight for the individual primer self binding value as in 
<a href="#PRIMER_MAX_SELF_ANY">PRIMER_MAX_SELF_ANY</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_SELF_ANY">PRIMER_INTERNAL_WT_SELF_ANY (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_SELF_ANY">PRIMER_WT_SELF_ANY</a> for the internal oligo.</p>


<h3><a name="PRIMER_PAIR_WT_COMPL_ANY">PRIMER_PAIR_WT_COMPL_ANY (float; default 0.0)</a></h3>

<p>Penalty weight for the binding value of the primer pair as in 
<a href="#PRIMER_MAX_SELF_ANY">PRIMER_MAX_SELF_ANY</a>.</p>


<h3><a name="PRIMER_MAX_SELF_END">PRIMER_MAX_SELF_END (decimal, 9999.99; default 3.00)</a></h3>

<p>
PRIMER_MAX_SELF_END tries to bind the 3'-END to a identical primer 
and scores the best binding it can find. This is critical for 
primer quality because it allows primers use itself as a target 
and amplify a short piece (forming a primer-dimer). These primers 
are then unable to bind and amplify the target sequence.<br>
PRIMER_MAX_SELF_END is the maximum allowable 3'-anchored global 
alignment score when testing a single primer for 
self-complementarity, and the maximum allowable 3'-anchored 
global alignment score when testing for complementarity between 
left and right primers. The 3'-anchored global alignment score 
is taken to predict the likelihood of PCR-priming primer-dimers, 
for example</p>
<pre>
   5' ATGCCCTAGCTTCCGGATG 3'
                ||| |||||
             3' AAGTCCTACATTTAGCCTAGT 5'
</pre>
or
<pre>
   5` AGGCTATGGGCCTCGCGA 3'
                  ||||||
               3' AGCGCTCCGGGTATCGGA 5'
</pre>
<p>The scoring system is as for the Maximum Complementarity
argument. In the examples above the scores are 7.00 and 6.00
respectively. Scores are non-negative, and a score of 0.00
indicates that there is no reasonable 3'-anchored global
alignment between two oligos.  In order to estimate 3'-anchored
global alignments for candidate primers and primer pairs, Primer
assumes that the sequence from which to choose primers is
presented 5'->3'. It is nonsensical to provide a larger value
for this parameter than for the Maximum (local) Complementarity
parameter (<a href="#PRIMER_MAX_SELF_ANY">PRIMER_MAX_SELF_ANY</a>) because the score of a local 
alignment will always be at least as great as the score of a 
global alignment.</p>


<h3><a name="PRIMER_INTERNAL_MAX_SELF_END">PRIMER_INTERNAL_MAX_SELF_END (decimal 9999.99; default 12.00)</a></h3>

<p><a href="#PRIMER_INTERNAL_MAX_SELF_END">PRIMER_INTERNAL_MAX_SELF_END</a> is meaningless when applied
to internal oligos used for hybridization-based detection, since
primer-dimer will not occur.  We recommend that
<a href="#PRIMER_INTERNAL_MAX_SELF_END">PRIMER_INTERNAL_MAX_SELF_END</a> be set at least as high as
<a href="#PRIMER_INTERNAL_MAX_SELF_ANY">PRIMER_INTERNAL_MAX_SELF_ANY</a>.</p>


<h3><a name="PRIMER_PAIR_MAX_COMPL_END">PRIMER_PAIR_MAX_COMPL_END (decimal, 9999.99; default 3.00)</a></h3>

<p>PRIMER_PAIR_MAX_COMPL_END tries to bind the 3'-END of the left primer 
to the right primer and scores the best binding it can find.
It is similar to <a href="#PRIMER_MAX_SELF_END">PRIMER_MAX_SELF_END</a>.</p>


<h3><a name="PRIMER_WT_SELF_END">PRIMER_WT_SELF_END (float; default 0.0)</a></h3>

<p>Penalty weight for the individual primer self binding value as in 
<a href="#PRIMER_MAX_SELF_END">PRIMER_MAX_SELF_END</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_SELF_END">PRIMER_INTERNAL_WT_SELF_END (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_SELF_END">PRIMER_WT_SELF_END</a> for the internal oligo.</p>


<h3><a name="PRIMER_PAIR_WT_COMPL_END">PRIMER_PAIR_WT_COMPL_END (float; default 0.0)</a></h3>

<p>Penalty weight for the binding value of the primer pair as in 
<a href="#PRIMER_MAX_SELF_END">PRIMER_MAX_SELF_END</a>.</p>


<h3><a name="PRIMER_MAX_END_STABILITY">PRIMER_MAX_END_STABILITY (float, 999.9999; default 100.0)</a></h3>

<p>The maximum stability for the last five 3' bases of a left or
right primer.  Bigger numbers mean more stable 3' ends.  The
value is the maximum delta G (kcal/mol) for duplex disruption for
the five 3' bases as calculated using the nearest-neighbor
parameter values specified by the option of <a href="#PRIMER_TM_SANTALUCIA">PRIMER_TM_SANTALUCIA</a> 
Table of thermodynamic parameters.<br>
<br>
For example if the table of thermodynamic parameters suggested
by <a href="http://dx.doi.org/10.1073/pnas.95.4.1460">SantaLucia 1998, DOI:10.1073/pnas.95.4.1460</a> is used the deltaG 
values for the most stable and for the most labile 5mer duplex 
are 6.86 kcal/mol (GCGCG) and 0.86 kcal/mol (TATAT) respectively.<br>
<br>	
If the table of thermodynamic parameters suggested by 
<a href="http://dx.doi.org/10.1073/pnas.83.11.3746">Breslauer et al. 1986, 10.1073/pnas.83.11.3746</a> is used the deltaG 
values for the most stable and for the most labile 5mer are 
13.4 kcal/mol (GCGCG) and 4.6 kcal/mol (TATAC) respectively.


<h3><a name="PRIMER_WT_END_STABILITY">PRIMER_WT_END_STABILITY (float; default 0.0)</a></h3>

<p>Penalty factor for the calculated maximum stability for the 
last five 3' bases of a left or right primer.</p>


<h3><a name="PRIMER_MAX_NS_ACCEPTED">PRIMER_MAX_NS_ACCEPTED (int; default 0)</a></h3>

<p>Maximum number of unknown bases (N) allowable in any primer.</p>


<h3><a name="PRIMER_INTERNAL_MAX_NS_ACCEPTED">PRIMER_INTERNAL_MAX_NS_ACCEPTED (int; default 0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MAX_NS_ACCEPTED">PRIMER_MAX_NS_ACCEPTED</a> for the internal oligo.</p>


<h3><a name="PRIMER_WT_NUM_NS">PRIMER_WT_NUM_NS (float; default 0.0)</a></h3>

<p>Penalty weight for the number of Ns in the primer.</p>


<h3><a name="PRIMER_INTERNAL_WT_NUM_NS">PRIMER_INTERNAL_WT_NUM_NS (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_NUM_NS">PRIMER_WT_NUM_NS</a> for the internal oligo.</p>


<h3><a name="PRIMER_MAX_POLY_X">PRIMER_MAX_POLY_X (int; default 5)</a></h3>

<p>The maximum allowable length of a mononucleotide repeat,
for example AAAAAA.</p>


<h3><a name="PRIMER_INTERNAL_MAX_POLY_X">PRIMER_INTERNAL_MAX_POLY_X (int; default 5)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MAX_POLY_X">PRIMER_MAX_POLY_X</a> for the internal oligo.</p>


<h3><a name="PRIMER_MIN_THREE_PRIME_DISTANCE">PRIMER_MIN_THREE_PRIME_DISTANCE (int; default -1)</a></h3>

<p>When returning multiple primer pairs,
the minimum number of base pairs between the 3' ends of
any two left or any two right primers.  The
objective is get 'truly different' primer pairs, where
the caller/user has some control over what 'truly different'
means.<br>
<br>
Primers with 3-prime ends at positions
e.g.  30 and 31 in the template
sequence have a three-prime distance of 1.<br>
<br>
In addition to positive values, the values -1 and 0 
are acceptable and have special interpretations:
<br>
<br>
-1 indicates that a given primer can appear in
multiple primer pairs returned by primer3
The only requirement is that each primer pair
be unique.  This is the default behavior and the behavior
in previous releases.  This is also the most liberal
behavior.<br>
<br>
0 indicates a primer pair is acceptable if neither the left or the right primer 
already appeared in any of the previous pairs.<br>
<br>
For n &gt; 0:
A primer pair is acceptable if:<br>
<br>
NOT(3' end of left primer closer than n to the 3' end a left primer in
an existing pair)<br>
<br>
AND<br>
<br>
NOT(3' end of right primer closer than n to the 3' end of right primer
in an existing pair)</p>


<h3><a name="PRIMER_PICK_ANYWAY">PRIMER_PICK_ANYWAY (boolean; default 0)</a></h3>

<p>If true use primer provided in <a href="#SEQUENCE_PRIMER">SEQUENCE_PRIMER</a>, <a href="#SEQUENCE_PRIMER_REVCOMP">SEQUENCE_PRIMER_REVCOMP</a>, 
or <a href="#SEQUENCE_INTERNAL_OLIGO">SEQUENCE_INTERNAL_OLIGO</a> even if it violates specific 
constraints.</p>


<h3><a name="PRIMER_LOWERCASE_MASKING">PRIMER_LOWERCASE_MASKING (int; default 0)</a></h3>

<p>This option allows for intelligent design of primers in sequence in
which masked regions (for example repeat-masked regions) are
lower-cased.  (New in v. 1.1.0, added by Maido Remm and Triinu
Koressaar)<br>
<br>
A value of 1 directs primer3 to reject primers overlapping
lowercase a base exactly at the 3' end.<br>
<br>
This property relies on the assumption that masked features
(e.g. repeats) can partly overlap primer, but they cannot overlap
the 3'-end of the primer.  In other words, lowercase bases at
other positions in the primer are accepted, assuming that the
masked features do not influence the primer performance if they
do not overlap the 3'-end of primer.</p>


<h3><a name="PRIMER_EXPLAIN_FLAG">PRIMER_EXPLAIN_FLAG (boolean; default 0)</a></h3>

<p>If this flag is 1 (non-0), produce <a href="#PRIMER_LEFT_EXPLAIN">PRIMER_LEFT_EXPLAIN</a>,
<a href="#PRIMER_RIGHT_EXPLAIN">PRIMER_RIGHT_EXPLAIN</a>, <a href="#PRIMER_INTERNAL_EXPLAIN">PRIMER_INTERNAL_EXPLAIN</a> and/or
<a href="#PRIMER_PAIR_EXPLAIN">PRIMER_PAIR_EXPLAIN</a> output
tags as appropriate.  
These output tags are intended to provide information on the number of
oligos and primer pairs that primer3 examined and counts of
the number discarded for various reasons.  If -format_output is
set similar information is produced in the user-oriented output.</p>


<h3><a name="PRIMER_LIBERAL_BASE">PRIMER_LIBERAL_BASE (boolean; default 0)</a></h3>

<p>This parameter provides a quick-and-dirty way to get primer3 to
accept IUB / IUPAC codes for ambiguous bases (i.e. by changing
all unrecognized bases to N).  If you wish to include an
ambiguous base in an oligo, you must set <a href="#PRIMER_NUM_NS_ACCEPTED">PRIMER_NUM_NS_ACCEPTED</a> to a
1 (non-0) value.<br>
<br>
Perhaps '-' and '* ' should be squeezed out rather than changed
to 'N', but currently they simply get converted to N's.  The authors
invite user comments.</p>


<h3><a name="PRIMER_FIRST_BASE_INDEX">PRIMER_FIRST_BASE_INDEX (int; default 0)</a></h3>

<p>This parameter is the index of the first base in the input
sequence.  For input and output using 1-based indexing (such as
that used in GenBank and to which many users are accustomed) set
this parameter to 1.  For input and output using 0-based indexing
set this parameter to 0.  (This parameter also affects the
indexes in the contents of the files produced when the primer
file flag is set.)</p>


<h3><a name="PRIMER_MAX_TEMPLATE_MISPRIMING">PRIMER_MAX_TEMPLATE_MISPRIMING (decimal, 9999.99; default -1.00)</a></h3>

<p>The maximum allowed similarity to ectopic sites in the
template.  A negative value means do not check.  The scoring
system is the same as used for <a href="#PRIMER_MAX_LIBRARY_MISPRIMING">PRIMER_MAX_LIBRARY_MISPRIMING</a>, except
that an ambiguity code in the template is never treated as a
consensus (see <a href="#PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS">PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS</a>).</p>


<h3><a name="PRIMER_INTERNAL_MAX_TEMPLATE_MISHYB">PRIMER_INTERNAL_MAX_TEMPLATE_MISHYB (decimal, 9999.99; default -1.00)</a></h3>

<p>Not implemented.</p>


<h3><a name="PRIMER_PAIR_MAX_TEMPLATE_MISPRIMING">PRIMER_PAIR_MAX_TEMPLATE_MISPRIMING (decimal, 9999.99; default -1.00)</a></h3>

<p>The maximum allowed summed similarity of both primers to
ectopic sites in the template. A negative value means do not
check.  The scoring system is the same as used for
<a href="#PRIMER_PAIR_MAX_LIBRARY_MISPRIMING">PRIMER_PAIR_MAX_LIBRARY_MISPRIMING</a>, except that an ambiguity code 
in the template is never treated as a consensus (see
<a href="#PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS">PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS</a>).  Primer3 does not
check the similarity of hybridization oligos (internal
oligos) to locations outside of the amplicon.</p>


<h3><a name="PRIMER_WT_TEMPLATE_MISPRIMING">PRIMER_WT_TEMPLATE_MISPRIMING (float; default 0.0)</a></h3>

<p>Penalty for a single primer binding to the template sequence.<br>
<br>
The use of this Tag is modified from primer3 version 2.0 on: 
The values used with the older versions have to be multiplied 
by the factor 100 to have the same effect.</p>


<h3><a name="PRIMER_INTERNAL_WT_TEMPLATE_MISHYB">PRIMER_INTERNAL_WT_TEMPLATE_MISHYB (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_TEMPLATE_MISPRIMING">PRIMER_WT_TEMPLATE_MISPRIMING</a> for the internal oligo.


<h3><a name="PRIMER_PAIR_WT_TEMPLATE_MISPRIMING">PRIMER_PAIR_WT_TEMPLATE_MISPRIMING (float; default 0.0)</a></h3>

<p>Penalty for a primer pair binding to the template sequence.<br>
<br>
The use of this Tag is modified from primer3 version 2.0 on: 
The values used with the older versions have to be multiplied 
by the factor 100 to have the same effect.</p>


<h3><a name="PRIMER_MISPRIMING_LIBRARY">PRIMER_MISPRIMING_LIBRARY (string; default empty)</a></h3>

<p>The name of a file containing a nucleotide sequence library of
sequences to avoid amplifying (for example repetitive sequences, or
possibly the sequences of genes in a gene family that should
not be amplified.)  The file must be in (a slightly restricted)
FASTA format (W. B. Pearson and D.J. Lipman, PNAS 85:8 pp
2444-2448 [1988]); we briefly discuss the organization of this
file below.  If this parameter is specified then primer3 locally
aligns each candidate primer against each library sequence and
rejects those primers for which the local alignment score times a
specified weight (see below) exceeds <a href="#PRIMER_MAX_LIBRARY_MISPRIMING">PRIMER_MAX_LIBRARY_MISPRIMING</a>.
(The maximum value of the weight is arbitrarily set to 100.0.)<br>
<br>
Each sequence entry in the FASTA-format file must begin with an
"id line" that starts with '>'.  The contents of the id line is
"slightly restricted" in that primer3 parses everything after any
optional asterisk ('*') as a floating point number to use as the
weight mentioned above.  If the id line contains no asterisk then
the weight defaults to 1.0.  The alignment scoring system used is
the same as for calculating complementarity among oligos (e.g.
<a href="#PRIMER_MAX_SELF_ANY">PRIMER_MAX_SELF_ANY</a>), except for the handling of IUB/IUPAC ambiguity
codes (discussed below).<br>
<br>
The remainder of an entry contains the sequence as lines
following the id line up until a line starting with '>' or
the end of the file.  Whitespace and newlines are ignored.
Characters 'A', 'T', 'G', 'C', 'a', 't', 'g', 'c' and
IUB/IUPAC 'ambiguity' codes ('R, 'Y', 'K', 'M', 'S', 'W',
'N', including lower case) are retained. For technical
reasons the length of the sequence must be >= 3. Of course,
sequences of length < 10 or so are probably useless, but
will be accepted without complaint.<br>
<br>
WARNING: always set <a href="#PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS">PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS</a>=0
if any sequence in the library contains strings of 'N's:
NNNNNNNNNNNNNNNNNNNN.<br>
NOWWW<br>
There are no restrictions on line length.<br>
<br>
An empty value for this parameter indicates that no repeat
library should be used and "turns off" the use of a
previously specified library.<br>
<br>
Repbase (J. Jurka, A.F.A. Smit, C. Pethiyagoda, and
others, 1995-1996, <a href="ftp://ncbi.nlm.nih.gov/repository/repbase">ftp://ncbi.nlm.nih.gov/repository/repbase</a>)
is an excellent source of repeat sequences and pointers to the
literature. (The Repbase files need to be converted to Fasta
format before they can be used by primer3.)</p>


<h3><a name="PRIMER_INTERNAL_MISHYB_LIBRARY">PRIMER_INTERNAL_MISHYB_LIBRARY (string; default empty)</a></h3>

<p>Similar to <a href="#PRIMER_MISPRIMING_LIBRARY">PRIMER_MISPRIMING_LIBRARY</a>, except that the event we
seek to avoid is hybridization of the internal oligo to sequences
in this library rather than priming from them.</p>


<h3><a name="PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS">PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS (boolean; default 1)</a></h3>

<p>If set to 1, treat ambiguity codes as if they were consensus
codes when matching oligos to mispriming or mishyb
libraries. For example, if this flag is set, then a C in an
oligo will be scored as a perfect match to an S in a library
sequence, as will a G in the oligo. More importantly,
though, any base in an oligo will be scored as a perfect
match to an N in the library.  This is very bad if the
library contains strings of Ns, as no oligo will be legal
(and it will take a long time to find this out). So unless
you know for sure that your library does not have runs of Ns
(or Xs), then set this flag to 0.</p>


<h3><a name="PRIMER_MAX_LIBRARY_MISPRIMING">PRIMER_MAX_LIBRARY_MISPRIMING (decimal, 9999.99; default 12.00)</a></h3>

<p>The maximum allowed weighted similarity with any sequence in
<a href="#PRIMER_MISPRIMING_LIBRARY">PRIMER_MISPRIMING_LIBRARY</a>.</p>


<h3><a name="PRIMER_INTERNAL_MAX_LIBRARY_MISHYB">PRIMER_INTERNAL_MAX_LIBRARY_MISHYB (decimal,9999.99; default 12.00)</a></h3>

<p>Similar to <a href="#PRIMER_MAX_LIBRARY_MISPRIMING">PRIMER_MAX_LIBRARY_MISPRIMING</a> except that this parameter applies
to the similarity of candidate internal oligos to the library
specified in <a href="#PRIMER_INTERNAL_MISHYB_LIBRARY">PRIMER_INTERNAL_MISHYB_LIBRARY</a>.</p>


<h3><a name="PRIMER_PAIR_MAX_LIBRARY_MISPRIMING">PRIMER_PAIR_MAX_LIBRARY_MISPRIMING (decimal, 9999.99; default 24.00)</a></h3>

<p>The maximum allowed sum of similarities of a primer pair
(one similarity for each primer) with any single sequence in
<a href="#PRIMER_MISPRIMING_LIBRARY">PRIMER_MISPRIMING_LIBRARY</a>.  
Library sequence weights are not used in computing the sum
of similarities.</p>


<h3><a name="PRIMER_WT_LIBRARY_MISPRIMING">PRIMER_WT_LIBRARY_MISPRIMING (float; default 0.0)</a></h3>

<p>Penalty for a single primer binding to any single sequence 
in <a href="#PRIMER_MISPRIMING_LIBRARY">PRIMER_MISPRIMING_LIBRARY</a>.</p>


<h3><a name="PRIMER_INTERNAL_WT_LIBRARY_MISHYB">PRIMER_INTERNAL_WT_LIBRARY_MISHYB (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_LIBRARY_MISPRIMING">PRIMER_WT_LIBRARY_MISPRIMING</a> for the internal oligo.</p>


<h3><a name="PRIMER_PAIR_WT_LIBRARY_MISPRIMING">PRIMER_PAIR_WT_LIBRARY_MISPRIMING (float; default 0.0)</a></h3>

<p>Penalty for a primer pair binding to any single sequence 
in <a href="#PRIMER_MISPRIMING_LIBRARY">PRIMER_MISPRIMING_LIBRARY</a>.</p>


<h3><a name="PRIMER_MIN_QUALITY">PRIMER_MIN_QUALITY (int; default 0)</a></h3>

<p>The minimum sequence quality (as specified by
<a href="#SEQUENCE_QUALITY">SEQUENCE_QUALITY</a>) allowed within a primer.</p>


<h3><a name="PRIMER_INTERNAL_MIN_QUALITY">PRIMER_INTERNAL_MIN_QUALITY (int; default 0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_MIN_QUALITY">PRIMER_MIN_QUALITY</a> for the internal oligo.</p>


<h3><a name="PRIMER_MIN_END_QUALITY">PRIMER_MIN_END_QUALITY (int; default 0)</a></h3>

<p>The minimum sequence quality (as specified by
<a href="#SEQUENCE_QUALITY">SEQUENCE_QUALITY</a>) allowed within the 5' pentamer of a
primer. Note that there is no <a href="#PRIMER_INTERNAL_MIN_END_QUALITY">PRIMER_INTERNAL_MIN_END_QUALITY</a>.</p>


<h3><a name="PRIMER_QUALITY_RANGE_MIN">PRIMER_QUALITY_RANGE_MIN (int; default 0)</a></h3>

<p>The minimum legal sequence quality (used for error checking
of <a href="#PRIMER_MIN_QUALITY">PRIMER_MIN_QUALITY</a> and <a href="#PRIMER_MIN_END_QUALITY">PRIMER_MIN_END_QUALITY</a>).</p>


<h3><a name="PRIMER_QUALITY_RANGE_MAX">PRIMER_QUALITY_RANGE_MAX (int; default 100)</a></h3>

<p>The maximum legal sequence quality (used for error checking
of <a href="#PRIMER_MIN_QUALITY">PRIMER_MIN_QUALITY</a> and <a href="#PRIMER_MIN_END_QUALITY">PRIMER_MIN_END_QUALITY</a>).</p>


<h3><a name="PRIMER_WT_SEQ_QUAL">PRIMER_WT_SEQ_QUAL (float; default 0.0)</a></h3>

<p>Penalty weight for the sequence quality of the primer.</p>


<h3><a name="PRIMER_INTERNAL_WT_SEQ_QUAL">PRIMER_INTERNAL_WT_SEQ_QUAL (float; default 0.0)</a></h3>

<p>Equivalent parameter of <a href="#PRIMER_WT_SEQ_QUAL">PRIMER_WT_SEQ_QUAL</a> for the internal oligo.</p>


<h3><a name="PRIMER_PAIR_WT_PR_PENALTY">PRIMER_PAIR_WT_PR_PENALTY (float; default 1.0)</a></h3>

<p>Penalty factor for the sum of the left and the right primer added 
to the pair penalty. Setting this value below 1.0 will increase 
running time.<br>
<br>
As <a href="#PRIMER_PAIR_WT_PR_PENALTY">PRIMER_PAIR_WT_PR_PENALTY</a> or the per-primer 
penalties it multiplies become lower with respect to various 
pair penalties (for example <a href="#PRIMER_PAIR_WT_PRODUCT_SIZE_LT">PRIMER_PAIR_WT_PRODUCT_SIZE_LT</a>
<a href="#PRIMER_PAIR_WT_PRODUCT_SIZE_GT">PRIMER_PAIR_WT_PRODUCT_SIZE_GT</a>
<a href="#PRIMER_PAIR_WT_DIFF_TM">PRIMER_PAIR_WT_DIFF_TM</a>, etc.) the running time of the 
search for primer pairs is likely to grow substantially. The 
reason is that the search algorithm must calculate the penalty 
for more primer pairs (as opposed to excluding them based on 
the penalties of the individual oligos).

</p>


<h3><a name="PRIMER_PAIR_WT_IO_PENALTY">PRIMER_PAIR_WT_IO_PENALTY (float; default 0.0)</a></h3>

<p>Penalty factor for the internal oligo added to the pair penalty.</p>


<h3><a name="PRIMER_INSIDE_PENALTY">PRIMER_INSIDE_PENALTY (float; default -1.0)</a></h3>

<p>Non-default values are valid only for sequences with 0 or 1
target regions.  If the primer is part of a pair that spans a
target and overlaps the target, then multiply this value times
the number of nucleotide positions by which the primer overlaps
the (unique) target to get the 'position penalty'.  The effect of
this parameter is to allow primer3 to include overlap with the
target as a term in the objective function.</p>


<h3><a name="PRIMER_OUTSIDE_PENALTY">PRIMER_OUTSIDE_PENALTY (float; default 0.0)</a></h3>

<p>Non-default values are valid only for sequences with 0 or 1
target regions.  If the primer is part of a pair that spans a
target and does not overlap the target, then multiply this value
times the number of nucleotide positions from the 3' end to the
(unique) target to get the 'position penalty'.  The effect of
this parameter is to allow primer3 to include nearness to the
target as a term in the objective function.</p>


<h3><a name="PRIMER_WT_POS_PENALTY">PRIMER_WT_POS_PENALTY (float; default 1.0)</a></h3>

<p>Penalty for the primer which do not overlap the target.</p>


<h3><a name="PRIMER_SEQUENCING_LEAD">PRIMER_SEQUENCING_LEAD (int; default 50)</a></h3>

<p>Defines the space from the 3'end of the primer to the point 
were the trace signals are readable. Value only used if 
<a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_sequencing_primers.</p>


<h3><a name="PRIMER_SEQUENCING_SPACING">PRIMER_SEQUENCING_SPACING (int; default 500)</a></h3>

<p>Defines the space from the 3'end of the primer to the 3'end of 
the next primer on the same strand. Value only used if 
<a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_sequencing_primers.</p>


<h3><a name="PRIMER_SEQUENCING_INTERVAL">PRIMER_SEQUENCING_INTERVAL (int; default 250)</a></h3>

<p>Defines the space from the 3'end of the primer to the 3'end of 
the next primer on the reverse strand. Value only used if 
<a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_sequencing_primers.</p>


<h3><a name="PRIMER_SEQUENCING_ACCURACY">PRIMER_SEQUENCING_ACCURACY (int; default 20)</a></h3>

<p>Defines the space from the calculated position of the 3'end to 
both sides in which primer3plus picks the best primer. Value only used if 
<a href="#PRIMER_TASK">PRIMER_TASK</a>=pick_sequencing_primers.</p>


<h3><a name="PRIMER_WT_END_QUAL">PRIMER_WT_END_QUAL (float; default 0.0)</a></h3>

<p> </p>


<h3><a name="PRIMER_INTERNAL_WT_END_QUAL">PRIMER_INTERNAL_WT_END_QUAL (float; default 0.0)</a></h3>

<p> </p>


<h2><a name="outputHelp">10. OUTPUT HELP</a></h2>

<h3><a name=p3w_primer_start>start (Start Position)</a></h3>
<p>The position of the 5' base of the primer.
For a Left Primer or Hyb Oligo this position is the position
of the leftmost base.  For a Right primer it is the position of
the <i><strong>rightmost</strong></i> base.</p>

<h3><a name=p3w_primer_len>len (Oligo Length)</a></h3>
<p>The length of the primer or oligo.</p>

<h3><a name=p3w_primer_tm>tm (Melting Temperature)</a></h3>
<p>The melting temperature of the primer or oligo.</p>

<h3><a name=p3w_primer_gc>gc%</strong></a></h2>
<p>The percent of G or C bases in the primer or oligo.</p>

<h3><a name=p3w_primer_any>any (Self Complementarity)</a></h3>
<p>The self-complementarity score of the oligo or primer (taken as
a measure of its tendency to anneal to itself or form
secondary structure).</p>

<h3><a name=p3w_primer_three>3' (Self Complementarity)</a></h3>
<p>The 3' self-complementarity of the primer or oligo (taken
as a measure of its tendency to form a primer-dimer with itself).</p>

<h3><a name=p3w_primer_repeat>rep (Mispriming or Mishyb Library Similarity)</a></h2>
<p>The similarity to the specified Mispriming or Mishyb library.</p>

<h3><a name=p3w_primer_seq>seq (Primer Sequence, 5' to 3')</a></h3>
<p>The sequence of the selected primer or oligo, always 5' to 3' so the
right primer is on the opposite strand from the one supplied in
the source input.  (The right primer sequence is the sequence
you would want synthesized in a primer order.)</p>

<h2><a name="providedMisprimingLibs">11. POVIDED MISPRIMING LIBRARIES</a></h2>

<p>This selection indicates what mispriming library (if any) Primer3 
should use to screen for interspersed repeats or for other 
sequence to avoid as a location for primers. The human and 
rodent libraries on the web page are adapted from Repbase 
(J. Jurka, A.F.A. Smit, C. Pethiyagoda, et al., 1995-1996)
<a href="ftp://ftp.ncbi.nih.gov/repository/repbase">ftp://ftp.ncbi.nih.gov/repository/repbase</a>).
The human library is humrep.ref concatenated with simple.ref,
translated to FASTA format. There are two rodent libraries. One 
is rodrep.ref translated to FASTA format, and the other is 
rodrep.ref concatenated with simple.ref, translated to FASTA 
format.</p>
<p>The <i>Drosophila</i> library is the concatenation of two libraries
from the <a href="http://www.fruitfly.org">Berkeley Drosophila Genome Project</a>:<br>
<blockquote>
1. A library of transposable elements
<a href="http://genomebiology.com/2002/3/12/research/0084">
<i>The transposable elements of the Drosophila melanogaster euchromatin - a genomics perspective</i>
J.S. Kaminker, C.M. Bergman, B. Kronmiller, J. Carlson, R. Svirskas, S. Patel, E. Frise, D.A. Wheeler, S.E. Lewis, G.M. Rubin, M. Ashburner and S.E. Celniker
Genome Biology (2002) 3(12):research0084.1-0084.20</a>, 
<a href="http://www.fruitfly.org/p_disrupt/datasets/ASHBURNER/D_mel_transposon_sequence_set.fasta">
http://www.fruitfly.org/p_disrupt/datasets/ASHBURNER/D_mel_transposon_sequence_set.fasta</a>
<br><br>
2. A library of repetitive DNA sequences <a href="http://www.fruitfly.org/sequence/sequence_db/na_re.dros">
http://www.fruitfly.org/sequence/sequence_db/na_re.dros</a>.
</blockquote>
Both were downloaded 6/23/04.<br>
<br>
The contents of the libraries can be viewed at the following links:<br>
<br>
<ul>
<li><a href="show_humrep_and_simple.cgi">HUMAN</a> (contains microsatellites)</li>
<li><a href="show_rodrep_and_simple.cgi">RODENT_AND_SIMPLE</a> (contains microsatellites)</li>
<li><a href="show_rodent_ref.cgi">RODENT</a> (does not contain microsatellites)</li>
<li><a href="show_drosophila_w_transposons.cgi">DROSOPHILA</a></li>
</ul>
<br></p>

<h2><a name="calculatePenalties">12. HOW PRIMER3 CALCULATES THE PENALTY VALUE</a></h2>

<p>
In essense, the penalty values define what is the best primer
pair.
The calculation of penalty values
takes into consideration penalty weights, which allow
one to fine-tune the selection of primers to specific needs.<br>
<br>
This section will explain the selection process of primers by 
primer3. In general the selection is a multi step process:<br>
<br>
In the first step, primer3 evaluates every primer that 
can be picked in the region of interest, possibly subject to
constraints due to target regions, product size ranges, and
so forth, that might preclude the use of primers in the
eventually selected primer pairs. In this pass the hard 
limits are tested like <a href="#PRIMER_MAX_GC">PRIMER_MAX_GC</a> or <a href="#PRIMER_MIN_TM">PRIMER_MIN_TM</a>. Primers 
with a GC lower than <a href="#PRIMER_MAX_GC">PRIMER_MAX_GC</a> or a Tm higher than 
<a href="#PRIMER_MIN_TM">PRIMER_MIN_TM</a> are memorized, the primers which fail in one of 
these tests are excluded. Primer3 can be forced to use primers 
failing to pass this test by setting <a href="#PRIMER_PICK_ANYWAY">PRIMER_PICK_ANYWAY</a> to one 
(only available for primers provided by the user).<br>
<br>
In the second step, primer3 calculates a penalty for each
primer. This penalty is the only score by which primer3 
evaluates the primers It is also provided as output 
<a href="#PRIMER_LEFT_4_PENALTY">PRIMER_LEFT_4_PENALTY</a>, <a href="#PRIMER_INTERNAL_4_PENALTY">PRIMER_INTERNAL_4_PENALTY</a> and 
<a href="#PRIMER_RIGHT_4_PENALTY">PRIMER_RIGHT_4_PENALTY</a> (shown for the fifth primer set). For each 
primer, it is calculated like that:</p>

<h3>Left Primers:</h3>
<pre>
<a href="#PRIMER_LEFT_4_PENALTY">PRIMER_LEFT_4_PENALTY</a> =

   <a href="#If PRIMER_LEFT_4_TM">If PRIMER_LEFT_4_TM</a> &gt; <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> then this is added (+):

       + <a href="#PRIMER_WT_TM_GT">PRIMER_WT_TM_GT</a> * ( <a href="#PRIMER_LEFT_4_TM">PRIMER_LEFT_4_TM</a> - <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> )

   If <a href="#PRIMER_LEFT_4_TM">PRIMER_LEFT_4_TM</a> &lt; <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> then this is added (+):

       + <a href="#PRIMER_WT_TM_LT">PRIMER_WT_TM_LT</a> * ( <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> - <a href="#PRIMER_LEFT_4_TM">PRIMER_LEFT_4_TM</a> )


   If <a href="#PRIMER_LEFT_4_GC_PERCENT">PRIMER_LEFT_4_GC_PERCENT</a> &gt; <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> then 
       this is added (+):

       + <a href="#PRIMER_WT_GC_PERCENT_GT">PRIMER_WT_GC_PERCENT_GT</a> *
            ( <a href="#PRIMER_LEFT_4_GC_PERCENT">PRIMER_LEFT_4_GC_PERCENT</a> - <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> )

   If <a href="#PRIMER_LEFT_4_GC_PERCENT">PRIMER_LEFT_4_GC_PERCENT</a> &lt; <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> then
       this is added (+):

       + <a href="#PRIMER_WT_GC_PERCENT_LT">PRIMER_WT_GC_PERCENT_LT</a> *
            ( <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> - <a href="#PRIMER_LEFT_4_GC_PERCENT">PRIMER_LEFT_4_GC_PERCENT</a> )


   The following section uses &lt;primer length&gt; as part of 
   the term which is given as output in 
   <a href="#PRIMER_LEFT_4">PRIMER_LEFT_4</a>=position,&lt;primer length&gt; 

   If &lt;primer length&gt; &gt; <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> then
       this is added (+):

       + <a href="#PRIMER_WT_SIZE_GT">PRIMER_WT_SIZE_GT</a> *
            ( &lt;primer length&gt; - <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> )

   If &lt;primer length&gt; &lt; <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> then
       this is added (+):

       + <a href="#PRIMER_WT_SIZE_LT">PRIMER_WT_SIZE_LT</a> *
            ( <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> - &lt;primer length&gt; )


   If the primer does not overlap a target then
       this is added (+):
	
       + <a href="#PRIMER_WT_POS_PENALTY">PRIMER_WT_POS_PENALTY</a> * <a href="#PRIMER_LEFT_4_POSITION_PENALTY">PRIMER_LEFT_4_POSITION_PENALTY</a>


   These are allways added (+) to the penalty:

       + <a href="#PRIMER_WT_SELF_ANY">PRIMER_WT_SELF_ANY</a> * <a href="#PRIMER_LEFT_4_SELF_ANY">PRIMER_LEFT_4_SELF_ANY</a>

       + <a href="#PRIMER_WT_SELF_END">PRIMER_WT_SELF_END</a> * <a href="#PRIMER_LEFT_4_SELF_END">PRIMER_LEFT_4_SELF_END</a>

       + <a href="#PRIMER_WT_TEMPLATE_MISPRIMING">PRIMER_WT_TEMPLATE_MISPRIMING</a> *
            <a href="#PRIMER_LEFT_4_TEMPLATE_MISPRIMING">PRIMER_LEFT_4_TEMPLATE_MISPRIMING</a>

       + <a href="#PRIMER_WT_END_STABILITY">PRIMER_WT_END_STABILITY</a> * <a href="#PRIMER_LEFT_4_END_STABILITY">PRIMER_LEFT_4_END_STABILITY</a>

       + <a href="#PRIMER_WT_NUM_NS">PRIMER_WT_NUM_NS</a> * &lt;numbers of N in the selected primer&gt;

       + <a href="#PRIMER_WT_LIBRARY_MISPRIMING">PRIMER_WT_LIBRARY_MISPRIMING</a> * <a href="#PRIMER_LEFT_4_LIBRARY_MISPRIMING">PRIMER_LEFT_4_LIBRARY_MISPRIMING</a>

       + <a href="#PRIMER_WT_SEQ_QUAL">PRIMER_WT_SEQ_QUAL</a> *
            ( <a href="#PRIMER_QUALITY_RANGE_MAX">PRIMER_QUALITY_RANGE_MAX</a> - 
              <a href="#PRIMER_LEFT_4_MIN_SEQ_QUALITY">PRIMER_LEFT_4_MIN_SEQ_QUALITY</a> )

</pre>

<h3>Right Primers (identical to Left Primers):</h3>
<pre>
<a href="#PRIMER_RIGHT_4_PENALTY">PRIMER_RIGHT_4_PENALTY</a> =

   If <a href="#PRIMER_RIGHT_4_TM">PRIMER_RIGHT_4_TM</a> &gt; <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> then 
       this is added (+):

       + <a href="#PRIMER_WT_TM_GT">PRIMER_WT_TM_GT</a> * ( <a href="#PRIMER_RIGHT_4_TM">PRIMER_RIGHT_4_TM</a> - <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> )

   If <a href="#PRIMER_RIGHT_4_TM">PRIMER_RIGHT_4_TM</a> &lt; <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> then 
       this is added (+):

       + <a href="#PRIMER_WT_TM_LT">PRIMER_WT_TM_LT</a> * ( <a href="#PRIMER_OPT_TM">PRIMER_OPT_TM</a> - <a href="#PRIMER_RIGHT_4_TM">PRIMER_RIGHT_4_TM</a> )


   If <a href="#PRIMER_RIGHT_4_GC_PERCENT">PRIMER_RIGHT_4_GC_PERCENT</a> &gt; <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> then 
       this is added (+):

       + <a href="#PRIMER_WT_GC_PERCENT_GT">PRIMER_WT_GC_PERCENT_GT</a> *
            ( <a href="#PRIMER_RIGHT_4_GC_PERCENT">PRIMER_RIGHT_4_GC_PERCENT</a> - <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> )

   If <a href="#PRIMER_RIGHT_4_GC_PERCENT">PRIMER_RIGHT_4_GC_PERCENT</a> &lt; <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> then 
       this is added (+):

       + <a href="#PRIMER_WT_GC_PERCENT_LT">PRIMER_WT_GC_PERCENT_LT</a> *
            ( <a href="#PRIMER_OPT_GC_PERCENT">PRIMER_OPT_GC_PERCENT</a> - <a href="#PRIMER_RIGHT_4_GC_PERCENT">PRIMER_RIGHT_4_GC_PERCENT</a> )


   The following section uses &lt;primer length&gt; as part of 
   the term which is given as output in 
   <a href="#PRIMER_RIGHT_4">PRIMER_RIGHT_4</a>=position,&lt;primer length&gt; 

   If &lt;primer length&gt; &gt; <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> then 
       this is added (+):

       + <a href="#PRIMER_WT_SIZE_GT">PRIMER_WT_SIZE_GT</a> *
            ( &lt;primer length&gt; - <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> )

   If &lt;primer length&gt; &lt; <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> then 
       this is added (+):

       + <a href="#PRIMER_WT_SIZE_LT">PRIMER_WT_SIZE_LT</a> *
            ( <a href="#PRIMER_OPT_SIZE">PRIMER_OPT_SIZE</a> - &lt;primer length&gt; )


   If the primer does not overlap a target then 
       this is added (+):

       + <a href="#PRIMER_WT_POS_PENALTY">PRIMER_WT_POS_PENALTY</a> * <a href="#PRIMER_RIGHT_4_POSITION_PENALTY">PRIMER_RIGHT_4_POSITION_PENALTY</a>


   These are allways added (+) to the penalty:

       + <a href="#PRIMER_WT_SELF_ANY">PRIMER_WT_SELF_ANY</a> * <a href="#PRIMER_RIGHT_4_SELF_ANY">PRIMER_RIGHT_4_SELF_ANY</a>

       + <a href="#PRIMER_WT_SELF_END">PRIMER_WT_SELF_END</a> * <a href="#PRIMER_RIGHT_4_SELF_END">PRIMER_RIGHT_4_SELF_END</a>

       + <a href="#PRIMER_WT_TEMPLATE_MISPRIMING">PRIMER_WT_TEMPLATE_MISPRIMING</a> *
            <a href="#PRIMER_RIGHT_4_TEMPLATE_MISPRIMING">PRIMER_RIGHT_4_TEMPLATE_MISPRIMING</a>

       + <a href="#PRIMER_WT_END_STABILITY">PRIMER_WT_END_STABILITY</a> * <a href="#PRIMER_RIGHT_4_END_STABILITY">PRIMER_RIGHT_4_END_STABILITY</a>

       + <a href="#PRIMER_WT_NUM_NS">PRIMER_WT_NUM_NS</a> * &lt;numbers of N in the selected primer&gt;

       + <a href="#PRIMER_WT_LIBRARY_MISPRIMING">PRIMER_WT_LIBRARY_MISPRIMING</a> * <a href="#PRIMER_RIGHT_4_LIBRARY_MISPRIMING">PRIMER_RIGHT_4_LIBRARY_MISPRIMING</a>

       + <a href="#PRIMER_WT_SEQ_QUAL">PRIMER_WT_SEQ_QUAL</a> *
            ( <a href="#PRIMER_QUALITY_RANGE_MAX">PRIMER_QUALITY_RANGE_MAX</a> - 
              <a href="#PRIMER_RIGHT_4_MIN_SEQ_QUALITY">PRIMER_RIGHT_4_MIN_SEQ_QUALITY</a> )

</pre>

<h3>Internal Oligos:</h3>
<pre>
<a href="#PRIMER_INTERNAL_4_PENALTY">PRIMER_INTERNAL_4_PENALTY</a> =

   If <a href="#PRIMER_INTERNAL_4_TM">PRIMER_INTERNAL_4_TM</a> &gt; <a href="#PRIMER_INTERNAL_OPT_TM">PRIMER_INTERNAL_OPT_TM</a> then
       this is added (+):

       + <a href="#PRIMER_INTERNAL_WT_TM_GT">PRIMER_INTERNAL_WT_TM_GT</a> *
            ( <a href="#PRIMER_INTERNAL_4_TM">PRIMER_INTERNAL_4_TM</a> - <a href="#PRIMER_INTERNAL_OPT_TM">PRIMER_INTERNAL_OPT_TM</a> )

   If <a href="#PRIMER_INTERNAL_4_TM">PRIMER_INTERNAL_4_TM</a> &lt; <a href="#PRIMER_INTERNAL_OPT_TM">PRIMER_INTERNAL_OPT_TM</a> then
       this is added (+):

       + <a href="#PRIMER_INTERNAL_WT_TM_LT">PRIMER_INTERNAL_WT_TM_LT</a> *
            ( <a href="#PRIMER_INTERNAL_OPT_TM">PRIMER_INTERNAL_OPT_TM</a> - <a href="#PRIMER_INTERNAL_4_TM">PRIMER_INTERNAL_4_TM</a> )


   If <a href="#PRIMER_INTERNAL_4_GC_PERCENT">PRIMER_INTERNAL_4_GC_PERCENT</a> &gt; <a href="#PRIMER_INTERNAL_OPT_GC_PERCENT">PRIMER_INTERNAL_OPT_GC_PERCENT</a>
       then this is added (+):

       + <a href="#PRIMER_INTERNAL_WT_GC_PERCENT_GT">PRIMER_INTERNAL_WT_GC_PERCENT_GT</a> *
            ( <a href="#PRIMER_INTERNAL_4_GC_PERCENT">PRIMER_INTERNAL_4_GC_PERCENT</a> - 
              <a href="#PRIMER_INTERNAL_OPT_GC_PERCENT">PRIMER_INTERNAL_OPT_GC_PERCENT</a> )

   If <a href="#PRIMER_INTERNAL_4_GC_PERCENT">PRIMER_INTERNAL_4_GC_PERCENT</a> &lt; <a href="#PRIMER_INTERNAL_OPT_GC_PERCENT">PRIMER_INTERNAL_OPT_GC_PERCENT</a>
       then this is added (+):

       + <a href="#PRIMER_INTERNAL_WT_GC_PERCENT_LT">PRIMER_INTERNAL_WT_GC_PERCENT_LT</a> *
            ( <a href="#PRIMER_INTERNAL_OPT_GC_PERCENT">PRIMER_INTERNAL_OPT_GC_PERCENT</a> -
              <a href="#PRIMER_INTERNAL_4_GC_PERCENT">PRIMER_INTERNAL_4_GC_PERCENT</a> )


   The following section uses &lt;primer length&gt; as part of 
   the term which is given as output in 
   <a href="#PRIMER_INTERNAL_4">PRIMER_INTERNAL_4</a>=position,&lt;primer length&gt; 

   If &lt;primer length&gt; &gt; <a href="#PRIMER_INTERNAL_OPT_SIZE">PRIMER_INTERNAL_OPT_SIZE</a> then 
       this is added (+):

       + <a href="#PRIMER_INTERNAL_WT_SIZE_GT">PRIMER_INTERNAL_WT_SIZE_GT</a> *
            ( &lt;primer length&gt; - <a href="#PRIMER_INTERNAL_OPT_SIZE">PRIMER_INTERNAL_OPT_SIZE</a> )

   If &lt;primer length&gt; &lt; <a href="#PRIMER_INTERNAL_OPT_SIZE">PRIMER_INTERNAL_OPT_SIZE</a> then
       this is added (+):

       + <a href="#PRIMER_INTERNAL_WT_SIZE_LT">PRIMER_INTERNAL_WT_SIZE_LT</a> *
            ( <a href="#PRIMER_INTERNAL_OPT_SIZE">PRIMER_INTERNAL_OPT_SIZE</a> - &lt;primer length&gt; )


   These are allways added (+) to the penalty:

     + <a href="#PRIMER_INTERNAL_WT_SELF_ANY">PRIMER_INTERNAL_WT_SELF_ANY</a> * <a href="#PRIMER_INTERNAL_4_SELF_ANY">PRIMER_INTERNAL_4_SELF_ANY</a>

     + <a href="#PRIMER_INTERNAL_WT_SELF_END">PRIMER_INTERNAL_WT_SELF_END</a> * <a href="#PRIMER_INTERNAL_4_SELF_END">PRIMER_INTERNAL_4_SELF_END</a>

     + <a href="#PRIMER_INTERNAL_WT_NUM_NS">PRIMER_INTERNAL_WT_NUM_NS</a> *
          &lt;numbers of N in the selected primer&gt;

     + <a href="#PRIMER_INTERNAL_WT_LIBRARY_MISHYB">PRIMER_INTERNAL_WT_LIBRARY_MISHYB</a> *
          <a href="#PRIMER_INTERNAL_4_LIBRARY_MISHYB">PRIMER_INTERNAL_4_LIBRARY_MISHYB</a>

     + <a href="#PRIMER_INTERNAL_WT_SEQ_QUAL">PRIMER_INTERNAL_WT_SEQ_QUAL</a> * 
          ( <a href="#PRIMER_QUALITY_RANGE_MAX">PRIMER_QUALITY_RANGE_MAX</a> - 
            <a href="#PRIMER_INTERNAL_4_MIN_SEQ_QUALITY">PRIMER_INTERNAL_4_MIN_SEQ_QUALITY</a> )

</pre>
<p>	
The primers are then sorted by penalty and primer3 tries to pick 
the primers with the lowest penalty. For the <a href="#PRIMER_TASK">PRIMER_TASK</a>
<tt><i>pick_primer_list</i></tt> or <tt><i>pick_sequencing_primers</i></tt> the selection ends at 
this point. If primer pairs have to be selected, a 
<a href="#PRIMER_PAIR_4_PENALTY">PRIMER_PAIR_4_PENALTY</a> is calculated:
<pre>
<a href="#PRIMER_PAIR_4_PENALTY">PRIMER_PAIR_4_PENALTY</a> =

   To the pair penalty are at first the single primer penalties 
   added (+):

       + <a href="#PRIMER_PAIR_WT_PR_PENALTY">PRIMER_PAIR_WT_PR_PENALTY</a> *
            ( <a href="#PRIMER_LEFT_4_PENALTY">PRIMER_LEFT_4_PENALTY</a> + <a href="#PRIMER_RIGHT_4_PENALTY">PRIMER_RIGHT_4_PENALTY</a> )


   If internal oligo is picked then this is added (+):

       + <a href="#PRIMER_PAIR_WT_IO_PENALTY">PRIMER_PAIR_WT_IO_PENALTY</a> * <a href="#PRIMER_INTERNAL_4_PENALTY">PRIMER_INTERNAL_4_PENALTY</a>
	

   If <a href="#PRIMER_PAIR_4_PRODUCT_TM">PRIMER_PAIR_4_PRODUCT_TM</a> &gt; <a href="#PRIMER_PRODUCT_OPT_TM">PRIMER_PRODUCT_OPT_TM</a> then
       this is added (+):

       + <a href="#PRIMER_PAIR_WT_PRODUCT_TM_GT">PRIMER_PAIR_WT_PRODUCT_TM_GT</a> *
            ( <a href="#PRIMER_PAIR_4_PRODUCT_TM">PRIMER_PAIR_4_PRODUCT_TM</a> - <a href="#PRIMER_PRODUCT_OPT_TM">PRIMER_PRODUCT_OPT_TM</a> )

   If <a href="#PRIMER_PAIR_4_PRODUCT_TM">PRIMER_PAIR_4_PRODUCT_TM</a> &lt; <a href="#PRIMER_PRODUCT_OPT_TM">PRIMER_PRODUCT_OPT_TM</a> then
       this is added (+):

       + <a href="#PRIMER_PAIR_WT_PRODUCT_TM_LT">PRIMER_PAIR_WT_PRODUCT_TM_LT</a> *
            ( <a href="#PRIMER_PRODUCT_OPT_TM">PRIMER_PRODUCT_OPT_TM</a> - <a href="#PRIMER_PAIR_4_PRODUCT_TM">PRIMER_PAIR_4_PRODUCT_TM</a> )


   If <a href="#PRIMER_PAIR_4_PRODUCT_SIZE">PRIMER_PAIR_4_PRODUCT_SIZE</a> &gt; <a href="#PRIMER_PRODUCT_OPT_SIZE">PRIMER_PRODUCT_OPT_SIZE</a> then
       this is added (+):

       + <a href="#PRIMER_PAIR_WT_PRODUCT_SIZE_GT">PRIMER_PAIR_WT_PRODUCT_SIZE_GT</a> *
            ( <a href="#PRIMER_PAIR_4_PRODUCT_SIZE">PRIMER_PAIR_4_PRODUCT_SIZE</a> - <a href="#PRIMER_PRODUCT_OPT_SIZE">PRIMER_PRODUCT_OPT_SIZE</a> )

   If <a href="#PRIMER_PAIR_4_PRODUCT_SIZE">PRIMER_PAIR_4_PRODUCT_SIZE</a> &lt; <a href="#PRIMER_PRODUCT_OPT_SIZE">PRIMER_PRODUCT_OPT_SIZE</a> then
       this is added (+):

       + <a href="#PRIMER_PAIR_WT_PRODUCT_SIZE_LT">PRIMER_PAIR_WT_PRODUCT_SIZE_LT</a> *
            ( <a href="#PRIMER_PRODUCT_OPT_SIZE">PRIMER_PRODUCT_OPT_SIZE</a> - <a href="#PRIMER_PAIR_4_PRODUCT_SIZE">PRIMER_PAIR_4_PRODUCT_SIZE</a> )


   These are allways added (+) to the penalty:

     + <a href="#PRIMER_PAIR_WT_DIFF_TM">PRIMER_PAIR_WT_DIFF_TM</a> * 
          &lt;difference in Tm between the left and the right primer&gt;

     + <a href="#PRIMER_PAIR_WT_COMPL_ANY">PRIMER_PAIR_WT_COMPL_ANY</a> * <a href="#PRIMER_PAIR_4_COMPL_ANY">PRIMER_PAIR_4_COMPL_ANY</a>

     + <a href="#PRIMER_PAIR_WT_COMPL_END">PRIMER_PAIR_WT_COMPL_END</a> * <a href="#PRIMER_PAIR_4_COMPL_END">PRIMER_PAIR_4_COMPL_END</a>

     + <a href="#PRIMER_PAIR_WT_LIBRARY_MISPRIMING">PRIMER_PAIR_WT_LIBRARY_MISPRIMING</a> * <a href="#PRIMER_PAIR_4_LIBRARY_MISPRIMING">PRIMER_PAIR_4_LIBRARY_MISPRIMING</a>

     + <a href="#PRIMER_PAIR_WT_TEMPLATE_MISPRIMING">PRIMER_PAIR_WT_TEMPLATE_MISPRIMING</a> *
          <a href="#PRIMER_PAIR_4_TEMPLATE_MISPRIMING">PRIMER_PAIR_4_TEMPLATE_MISPRIMING</a>

</pre>
<p>Primer3 tries to select pairs with the lowest penalty which 
still fulfill all necessary requirements like non-redundancy or
product size limits.</p>

<h2><a name="webInterface">13. THE PRIMER3 WWW INTERFACE</a></h2>

<p>There are two web interfaces available :<br>
The Whitehead Institute for Biomedical Research provides a basic
web-based front end to primer3 named primer3web at
<a href="http://fokker.wi.mit.edu/cgi-bin/primer3/primer3_www.cgi">http://fokker.wi.mit.edu/cgi-bin/primer3/primer3_www.cgi</a><br>
<br>
The Wageningen University Laboratory of Bioinformatics provides a enhanced 
web-based front end to primer3 named primer3plus at
<a href="http://www.bioinformatics.nl/cgi-bin/primer3plus/primer3plus.cgi">http://www.bioinformatics.nl/cgi-bin/primer3plus/primer3plus.cgi</a><br>
<br>
This distribution does not contain the Primer3 WWW interfaces.<br>
<br>
Web interface code is available at
<a href="http://sourceforge.net/projects/primer3/">http://sourceforge.net/projects/primer3/</a>.</p>

<h2><a name="acknowledgments">14. ACKNOWLEDGMENTS</a></h2>

<p>Initial development of Primer3 was funded by Howard Hughes Medical
Institute and by the National Institutes of Health, National Human
Genome Research Institute under grants R01-HG00257 (to David C. Page)
and P50-HG00098 (to Eric S. Lander).<br>
<br>
Primer3 was originally written by Helen J. Skaletsky (Howard Hughes
Medical Institute, Whitehead Institute) and Steve Rozen (Duke-NUS
Graduate Medical School Singapore, formerly Whitehead Institute)
based on the design of earlier versions, notably Primer 0.5
(Steve Lincoln, Mark Daly, and Eric S. Lander).
The original web interface was designed by Richard Resnick.  Lincoln
Stein designed Boulder-IO format in the days before XML and RDF, and
championed the idea of making primer3 a software component, which
has been key to its wide utility.

In addition, among others, Ernst Molitor, Carl Foeller, and James Bonfield
contributed to the early
design of primer3. Brant Faircloth has helped with 
ensuring that primer3 runs on Windows and MacOS and with the
primer3 web site. 
Triinu Koressaar and Maido Remm modernized the melting 
temperature calculations.

<br>
<br>
Primer3 is now operating as an open software development project hosted
on SourceForge, and we are working out how to acknowledge all who have
contributed to its enhancement.  Current active developers can be
found at http://sourceforge.net/projects/primer3/.</p>

</div>
</body>
</html>
